// @fluentui/react-northstar/dist/dts/src/utils/commonPropTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.commonPropTypes = void 0;
var tslib_1 = require("tslib");
var customPropTypes = require("@fluentui/react-proptypes");
var PropTypes = require("prop-types");
exports.commonPropTypes = {
    createCommon: function (config) {
        if (config === void 0) { config = {}; }
        var _a = config.accessibility, accessibility = _a === void 0 ? true : _a, _b = config.as, as = _b === void 0 ? true : _b, _c = config.children, children = _c === void 0 ? 'node' : _c, _d = config.className, className = _d === void 0 ? true : _d, _e = config.color, color = _e === void 0 ? false : _e, _f = config.content, content = _f === void 0 ? 'node' : _f, _g = config.styled, styled = _g === void 0 ? true : _g;
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, (accessibility && {
            accessibility: customPropTypes.accessibility,
        })), (as && {
            as: PropTypes.elementType,
        })), (children && {
            children: children === 'element' ? PropTypes.element : PropTypes.node,
        })), (className && {
            className: PropTypes.string,
        })), (color && {
            color: PropTypes.string,
        })), (content && {
            content: content === 'shorthand' ? customPropTypes.itemShorthand : customPropTypes.nodeContent,
        })), (styled && {
            styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
            variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
            design: customPropTypes.design,
        }));
    },
};
