// @fluentui/react-northstar/dist/dts/src/components/Reaction/ReactionGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReactionGroup = exports.reactionGroupClassName = void 0;
var tslib_1 = require("tslib");
var customPropTypes = require("@fluentui/react-proptypes");
var React = require("react");
var _ = require("lodash");
var utils_1 = require("../../utils");
var Reaction_1 = require("./Reaction");
var react_bindings_1 = require("@fluentui/react-bindings");
exports.reactionGroupClassName = 'ui-reactions';
/**
 * A ReactionGroup groups multiple Reaction elements.
 */
var ReactionGroup = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.ReactionGroup.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var children = props.children, items = props.items, content = props.content, className = props.className, design = props.design, styles = props.styles, variables = props.variables;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.ReactionGroup.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: exports.ReactionGroup.displayName,
        rtl: context.rtl,
    });
    var _b = react_bindings_1.useStyles(exports.ReactionGroup.displayName, {
        className: exports.reactionGroupClassName,
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }), classes = _b.classes, resolvedStyles = _b.styles;
    var element = _.isNil(items) ? (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps)), utils_1.rtlTextContainer.getAttributes({ forElements: [children, content] })), utils_1.childrenExist(children) ? children : content)) : (React.createElement(ElementType, tslib_1.__assign({}, unhandledProps, { className: classes.root }), _.map(items, function (reaction) {
        return Reaction_1.Reaction.create(reaction, {
            defaultProps: function () { return ({
                styles: resolvedStyles.reaction,
            }); },
        });
    })));
    setEnd();
    return element;
};
exports.ReactionGroup = ReactionGroup;
exports.ReactionGroup.displayName = 'ReactionGroup';
exports.ReactionGroup.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { items: customPropTypes.collectionShorthand });
exports.ReactionGroup.handledProps = Object.keys(exports.ReactionGroup.propTypes);
exports.ReactionGroup.create = utils_1.createShorthandFactory({
    Component: exports.ReactionGroup,
    mappedProp: 'content',
    mappedArrayProp: 'items',
});
