// @fluentui/react-northstar/dist/dts/src/components/Reaction/Reaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Reaction = exports.reactionSlotClassNames = exports.reactionClassName = void 0;
var tslib_1 = require("tslib");
var customPropTypes = require("@fluentui/react-proptypes");
var React = require("react");
var utils_1 = require("../../utils");
var Box_1 = require("../Box/Box");
var ReactionGroup_1 = require("./ReactionGroup");
var react_bindings_1 = require("@fluentui/react-bindings");
exports.reactionClassName = 'ui-reaction';
exports.reactionSlotClassNames = {
    icon: exports.reactionClassName + "__icon",
    content: exports.reactionClassName + "__content",
};
/**
 * A Reaction indicates user's emotion or perception.
 * Used to display user's reaction for entity in Chat (e.g. message).
 */
var Reaction = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.Reaction.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var children = props.children, icon = props.icon, content = props.content, className = props.className, design = props.design, styles = props.styles, variables = props.variables;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.Reaction.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: exports.Reaction.displayName,
        rtl: context.rtl,
    });
    var _b = react_bindings_1.useStyles(exports.Reaction.displayName, {
        className: exports.reactionClassName,
        mapPropsToStyles: function () { return ({
            hasContent: !!content,
        }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }), classes = _b.classes, resolvedStyles = _b.styles;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps)), utils_1.rtlTextContainer.getAttributes({ forElements: [children] })), utils_1.childrenExist(children) ? (children) : (React.createElement(React.Fragment, null,
        Box_1.Box.create(icon, {
            defaultProps: function () {
                return getA11yProps('icon', {
                    className: exports.reactionSlotClassNames.icon,
                    styles: resolvedStyles.icon,
                });
            },
        }),
        Box_1.Box.create(content, {
            defaultProps: function () {
                return getA11yProps('content', {
                    className: exports.reactionSlotClassNames.content,
                    styles: resolvedStyles.content,
                });
            },
        })))));
    setEnd();
    return element;
};
exports.Reaction = Reaction;
exports.Reaction.displayName = 'Reaction';
exports.Reaction.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({
    content: 'shorthand',
})), { icon: customPropTypes.shorthandAllowingChildren });
exports.Reaction.defaultProps = {
    as: 'span',
};
exports.Reaction.handledProps = Object.keys(exports.Reaction.propTypes);
exports.Reaction.Group = ReactionGroup_1.ReactionGroup;
exports.Reaction.create = utils_1.createShorthandFactory({ Component: exports.Reaction, mappedProp: 'content' });
