// @fluentui/react-northstar/dist/dts/src/themes/teams/getBorderFocusStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBorderFocusStyles = void 0;
var tslib_1 = require("tslib");
var defaultColor = 'transparent';
/**
 * Returns style object that can be used for styling components on focus state.
 * NOTE: the element where this is used needs to have relative positioning so that the
 * pseudo elements created on focus can be properly positioned.
 */
var getBorderFocusStyles = function (args) {
    var _a, _b, _c, _d;
    var sv = args.variables;
    var _e = args.borderWidth, borderWidth = _e === void 0 ? sv.borderWidth : _e, _f = args.borderRadius, borderRadius = _f === void 0 ? sv.borderRadius : _f, _g = args.focusInnerBorderColor, focusInnerBorderColor = _g === void 0 ? sv.focusInnerBorderColor || defaultColor : _g, _h = args.focusOuterBorderColor, focusOuterBorderColor = _h === void 0 ? sv.focusOuterBorderColor || defaultColor : _h, borderPadding = args.borderPadding;
    var defaultPreudoStyles = {
        content: '""',
        position: 'absolute',
        borderStyle: 'solid',
        pointerEvents: 'none',
        borderWidth: borderWidth,
        borderRadius: borderRadius,
    };
    var borderPaddingTop = ((_a = borderPadding) === null || _a === void 0 ? void 0 : _a.top) || borderPadding;
    var borderPaddingBottom = ((_b = borderPadding) === null || _b === void 0 ? void 0 : _b.bottom) || borderPadding;
    var borderPaddingLeft = ((_c = borderPadding) === null || _c === void 0 ? void 0 : _c.left) || borderPadding;
    var borderPaddingRight = ((_d = borderPadding) === null || _d === void 0 ? void 0 : _d.right) || borderPadding;
    return {
        ':focus': {
            outline: 'none',
        },
        ':focus-visible': {
            borderColor: 'transparent',
            ':before': tslib_1.__assign(tslib_1.__assign({}, defaultPreudoStyles), { zIndex: sv.zIndexes.foreground, borderColor: focusInnerBorderColor, top: borderPadding == null ? '0' : "-" + borderPaddingTop, bottom: borderPadding == null ? '0' : "-" + borderPaddingBottom, left: borderPadding == null ? '0' : "-" + borderPaddingLeft, right: borderPadding == null ? '0' : "-" + borderPaddingRight }),
            ':after': tslib_1.__assign(tslib_1.__assign({}, defaultPreudoStyles), { zIndex: sv.zIndexes.foreground, borderColor: focusOuterBorderColor, top: borderPadding == null ? "-" + borderWidth : "calc(0px - " + borderPaddingTop + " - " + borderWidth + ")", bottom: borderPadding == null ? "-" + borderWidth : "calc(0px - " + borderPaddingBottom + " - " + borderWidth + ")", left: borderPadding == null ? "-" + borderWidth : "calc(0px - " + borderPaddingLeft + " - " + borderWidth + ")", right: borderPadding == null ? "-" + borderWidth : "calc(0px - " + borderPaddingRight + " - " + borderWidth + ")" }),
        },
    };
};
exports.getBorderFocusStyles = getBorderFocusStyles;
