// react-cosmos-shared2/dist/fixtureState/createValues.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createValues = createValues;
exports.createValue = createValue;

var _cjs = _interopRequireDefault(require("react-element-to-jsx-string/dist/cjs"));

var _reactIs = require("react-is");

var _shared = require("./shared");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// XXX: Importing the CJS dist works with both webpack and Snowpack. By default,
// Snowpack imports the ESM version of react-element-to-jsx-string. Because of
// how Babel compiles this file (createValues.ts), the default export of
// reactElementToJSXString ends up wrapped twice as
//   { default: { default: reactElementToJSXString } }
// I'm not sure what's the underlying issue, and if Snowpack, Babel or
// react-element-to-jsx-string is to blame.
function createValues(obj) {
  var values = {};
  Object.keys(obj).forEach(function (key) {
    values[key] = createValue(obj[key]);
  });
  return values;
}

function createValue(data) {
  if ((0, _shared.isPrimitiveData)(data)) return {
    type: 'primitive',
    data: data
  };
  if ((0, _shared.isArray)(data)) return {
    type: 'array',
    values: data.map(function (i) {
      return createValue(i);
    })
  };
  if ((0, _shared.isObject)(data)) return {
    type: 'object',
    values: createValues(data)
  }; // Why store unserializable values in fixture state?
  // - Because they still provides value in the Cosmos UI. They let the user know
  //   that, eg. a prop, is present and see the read-only stringified value.
  // - More importantly, because the fixture state controls which props to render.
  //   This way, if a prop is read-only and cannot be edited in the UI, it can
  //   still be removed.

  return {
    type: 'unserializable',
    stringifiedData: stringifyUnserializableData(data)
  };
}

function stringifyUnserializableData(data) {
  // TODO: Enable custom stringifier plugins
  return (0, _reactIs.isElement)(data) ? (0, _cjs.default)(data) : String(data);
}
