// react-cosmos-shared2/dist/fixtureState/props.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getFixtureStateProps = getFixtureStateProps;
exports.findFixtureStateProps = findFixtureStateProps;
exports.createFixtureStateProps = createFixtureStateProps;
exports.resetFixtureStateProps = resetFixtureStateProps;
exports.updateFixtureStateProps = updateFixtureStateProps;
exports.removeFixtureStateProps = removeFixtureStateProps;
exports.DEFAULT_RENDER_KEY = void 0;

var _lodash = require("lodash");

var _util = require("../util");

var DEFAULT_RENDER_KEY = 0;
exports.DEFAULT_RENDER_KEY = DEFAULT_RENDER_KEY;

function getFixtureStateProps(fixtureState, decoratorId) {
  var props = fixtureState.props;
  return props ? props.filter(function (p) {
    return p.elementId.decoratorId === decoratorId;
  }) : [];
}

function findFixtureStateProps(fixtureState, elementId) {
  var props = fixtureState.props;
  return props && (0, _lodash.find)(props, function (p) {
    return (0, _lodash.isEqual)(p.elementId, elementId);
  });
}

function createFixtureStateProps(_ref) {
  var fixtureState = _ref.fixtureState,
      elementId = _ref.elementId,
      values = _ref.values,
      componentName = _ref.componentName;
  var _fixtureState$props = fixtureState.props,
      props = _fixtureState$props === void 0 ? [] : _fixtureState$props;
  return (0, _util.replaceOrAddItem)(props, createPropsMatcher(elementId), {
    elementId: elementId,
    values: values,
    renderKey: DEFAULT_RENDER_KEY,
    componentName: componentName
  });
}

function resetFixtureStateProps(_ref2) {
  var fixtureState = _ref2.fixtureState,
      elementId = _ref2.elementId,
      values = _ref2.values;
  var propsItem = expectFixtureStateProps(fixtureState, elementId);
  return (0, _util.updateItem)(fixtureState.props, propsItem, {
    values: values,
    renderKey: propsItem.renderKey + 1
  });
}

function updateFixtureStateProps(_ref3) {
  var fixtureState = _ref3.fixtureState,
      elementId = _ref3.elementId,
      values = _ref3.values;
  var propsItem = expectFixtureStateProps(fixtureState, elementId);
  return (0, _util.updateItem)(fixtureState.props, propsItem, {
    values: values
  });
}

function removeFixtureStateProps(fixtureState, elementId) {
  return (0, _util.removeItemMatch)(fixtureState.props || [], createPropsMatcher(elementId));
}

function createPropsMatcher(elementId) {
  return function (p) {
    return (0, _lodash.isEqual)(p.elementId, elementId);
  };
}

function expectFixtureStateProps(fixtureState, elementId) {
  var propsItem = findFixtureStateProps(fixtureState, elementId);

  if (!propsItem) {
    var elId = JSON.stringify(elementId);
    throw new Error("Fixture state props missing for element \"".concat(elId, "\""));
  }

  return propsItem;
}
