// react-cosmos-shared2/dist/fixtureState/shared.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isString = isString;
exports.isNumber = isNumber;
exports.isBoolean = isBoolean;
exports.isNull = isNull;
exports.isUndefined = isUndefined;
exports.isPrimitiveData = isPrimitiveData;
exports.isObject = isObject;
exports.isArray = isArray;

var _lodash = require("lodash");

var _reactIs = require("react-is");

function isString(value) {
  return typeof value === 'string';
}

function isNumber(value) {
  return typeof value === 'number';
}

function isBoolean(value) {
  return typeof value === 'boolean';
}

function isNull(value) {
  return value === null;
}

function isUndefined(value) {
  return value === undefined;
}

function isPrimitiveData(data) {
  return isString(data) || isNumber(data) || isBoolean(data) || isNull(data) || isUndefined(data);
}

function isObject(data) {
  return (0, _lodash.isPlainObject)(data) && !(0, _reactIs.isElement)(data);
}

function isArray(data) {
  return Array.isArray(data);
}
