// @fluentui/react-northstar/dist/dts/src/components/Header/HeaderDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeaderDescription = exports.headerDescriptionClassName = void 0;
var tslib_1 = require("tslib");
var React = require("react");
var utils_1 = require("../../utils");
var react_bindings_1 = require("@fluentui/react-bindings");
exports.headerDescriptionClassName = 'ui-header__description';
/**
 * A HeaderDescription provides more detailed information about the Header.
 */
var HeaderDescription = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.HeaderDescription.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var children = props.children, content = props.content, color = props.color, className = props.className, design = props.design, styles = props.styles, variables = props.variables;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.HeaderDescription.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(props.accessibility, {
        debugName: exports.HeaderDescription.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(exports.HeaderDescription.displayName, {
        className: exports.headerDescriptionClassName,
        mapPropsToStyles: function () { return ({
            color: color,
        }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign(tslib_1.__assign({ className: classes.root }, unhandledProps), utils_1.rtlTextContainer.getAttributes({ forElements: [children, content] })))), utils_1.childrenExist(children) ? children : content));
    setEnd();
    return element;
};
exports.HeaderDescription = HeaderDescription;
exports.HeaderDescription.displayName = 'HeaderDescription';
exports.HeaderDescription.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({ color: true }));
exports.HeaderDescription.defaultProps = {
    as: 'p',
};
exports.HeaderDescription.handledProps = Object.keys(exports.HeaderDescription.propTypes);
exports.HeaderDescription.create = utils_1.createShorthandFactory({
    Component: exports.HeaderDescription,
    mappedProp: 'content',
});
