// reakit/es/Grid/GridCell.js
import { a as _objectSpread2 } from '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import { createComponent } from 'reakit-system/createComponent';
import { createHook } from 'reakit-system/createHook';
import 'reakit-utils/misc';
import 'react';
import 'reakit-utils/hooks';
import 'reakit-utils/dom';
import 'reakit-warning';
import 'reakit-utils/events';
import 'reakit-utils/focus';
import '../Role/Role.js';
import '../Tabbable/Tabbable.js';
import '../Clickable/Clickable.js';
import '../getCurrentId-5aa9849e.js';
import '../__keys-6742f591.js';
import '../userFocus-e16425e3.js';
import '../Id/IdProvider.js';
import '../Id/Id.js';
import '../setTextFieldValue-5c13f3b6.js';
import { useCompositeItem } from '../Composite/CompositeItem.js';
import { a as GRID_CELL_KEYS } from '../__keys-08a69d36.js';

var unstable_useGridCell = createHook({
  name: "GridCell",
  compose: useCompositeItem,
  keys: GRID_CELL_KEYS,
  useProps: function useProps(_, htmlProps) {
    return _objectSpread2({
      role: "gridcell"
    }, htmlProps);
  }
});
var unstable_GridCell = createComponent({
  as: "span",
  memo: true,
  useHook: unstable_useGridCell
});

export { unstable_GridCell, unstable_useGridCell };
