// react-cosmos/dist/shared/userDeps/generateUserDepsModule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateUserDepsModule = generateUserDepsModule;

var _findUserModulePaths2 = require("./findUserModulePaths");

var _userDepsTemplate = require("./userDepsTemplate");

function generateUserDepsModule(_ref) {
  var cosmosConfig = _ref.cosmosConfig,
      rendererConfig = _ref.rendererConfig,
      relativeToDir = _ref.relativeToDir;
  var rootDir = cosmosConfig.rootDir,
      fixturesDir = cosmosConfig.fixturesDir,
      fixtureFileSuffix = cosmosConfig.fixtureFileSuffix,
      globalImports = cosmosConfig.globalImports;

  var _findUserModulePaths = (0, _findUserModulePaths2.findUserModulePaths)({
    rootDir: rootDir,
    fixturesDir: fixturesDir,
    fixtureFileSuffix: fixtureFileSuffix
  }),
      fixturePaths = _findUserModulePaths.fixturePaths,
      decoratorPaths = _findUserModulePaths.decoratorPaths;

  return (0, _userDepsTemplate.userDepsTemplate)({
    globalImports: globalImports,
    fixturePaths: fixturePaths,
    decoratorPaths: decoratorPaths,
    rendererConfig: rendererConfig,
    rootDir: rootDir,
    relativeToDir: relativeToDir
  });
}
