// reakit/es/Tab/TabState.js
import { _ as _objectWithoutPropertiesLoose, a as _objectSpread2 } from '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import 'reakit-utils/misc';
import { useState, useCallback, useEffect } from 'react';
import { useInitialValue } from 'reakit-utils/hooks';
import 'reakit-utils/dom';
import '../reverse-30eaa122.js';
import '../getCurrentId-5aa9849e.js';
import '../findEnabledItemById-8ddca752.js';
import '../Id/IdProvider.js';
import '../Id/IdState.js';
import { useCompositeState } from '../Composite/CompositeState.js';

function useTabState(initialState) {
  if (initialState === void 0) {
    initialState = {};
  }

  var _useInitialValue = useInitialValue(initialState),
      initialSelectedId = _useInitialValue.selectedId,
      _useInitialValue$loop = _useInitialValue.loop,
      loop = _useInitialValue$loop === void 0 ? true : _useInitialValue$loop,
      _useInitialValue$manu = _useInitialValue.manual,
      manual = _useInitialValue$manu === void 0 ? false : _useInitialValue$manu,
      sealed = _objectWithoutPropertiesLoose(_useInitialValue, ["selectedId", "loop", "manual"]);

  var composite = useCompositeState(_objectSpread2({
    loop: loop,
    currentId: initialSelectedId
  }, sealed));
  var panels = useCompositeState();

  var _React$useState = useState(initialSelectedId),
      selectedId = _React$useState[0],
      setSelectedId = _React$useState[1];

  var select = useCallback(function (id) {
    composite.move(id);
    setSelectedId(id);
  }, [composite.move]); // If selectedId is not set, use the currentId. It's still possible to have
  // no selected tab with useTabState({ selectedId: null });

  useEffect(function () {
    if (selectedId === null) return;
    var selectedItem = composite.items.find(function (item) {
      return item.id === selectedId;
    });
    if (selectedItem) return;

    if (composite.currentId) {
      setSelectedId(composite.currentId);
    }
  }, [selectedId, composite.items, composite.currentId]);
  return _objectSpread2(_objectSpread2({}, composite), {}, {
    selectedId: selectedId,
    panels: panels.items,
    manual: manual,
    select: select,
    setSelectedId: setSelectedId,
    registerPanel: useCallback(function (panel) {
      return panels.registerItem(panel);
    }, [panels.registerItem]),
    unregisterPanel: useCallback(function (id) {
      return panels.unregisterItem(id);
    }, [panels.unregisterItem])
  });
}

export { useTabState };
