// @fluentui/react-northstar/dist/commonjs/components/Attachment/AttachmentAction.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.AttachmentAction = exports.attachmentActionClassName = void 0;

var _accessibility = require("@fluentui/accessibility");

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _Button = require("../Button/Button");

var attachmentActionClassName = 'ui-attachment__action';
/**
 * An AttachmentAction provides a slot for actions in the Attachment.
 */

exports.attachmentActionClassName = attachmentActionClassName;
var AttachmentAction = (0, _reactBindings.compose)(_Button.Button, {
  className: attachmentActionClassName,
  displayName: 'AttachmentAction',
  overrideStyles: true,
  shorthandConfig: {
    mappedProp: 'content'
  }
});
exports.AttachmentAction = AttachmentAction;
AttachmentAction.defaultProps = {
  accessibility: _accessibility.buttonBehavior,
  as: 'button'
};
AttachmentAction.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon({
  content: 'shorthand'
}), {
  disabled: PropTypes.bool,
  icon: customPropTypes.shorthandAllowingChildren,
  iconPosition: PropTypes.oneOf(['before', 'after']),
  loader: customPropTypes.itemShorthandWithoutJSX,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
  onFocus: PropTypes.func,
  primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
  secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool])
});
