// reakit/es/Toolbar/ToolbarItem.js
import '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import { createComponent } from 'reakit-system/createComponent';
import { createHook } from 'reakit-system/createHook';
import 'reakit-utils/misc';
import 'react';
import 'reakit-utils/hooks';
import 'reakit-utils/dom';
import 'reakit-warning';
import 'reakit-utils/events';
import 'reakit-utils/focus';
import '../Role/Role.js';
import '../Tabbable/Tabbable.js';
import '../Clickable/Clickable.js';
import '../getCurrentId-5aa9849e.js';
import '../__keys-6742f591.js';
import '../userFocus-e16425e3.js';
import '../Id/IdProvider.js';
import '../Id/Id.js';
import '../setTextFieldValue-5c13f3b6.js';
import { useCompositeItem } from '../Composite/CompositeItem.js';
import { a as TOOLBAR_ITEM_KEYS } from '../__keys-ae468c11.js';

var useToolbarItem = createHook({
  name: "ToolbarItem",
  compose: useCompositeItem,
  keys: TOOLBAR_ITEM_KEYS
});
var ToolbarItem = createComponent({
  as: "button",
  memo: true,
  useHook: useToolbarItem
});

export { ToolbarItem, useToolbarItem };
