// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Divider/dividerStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dividerStyles = void 0;
var tslib_1 = require("tslib");
var _ = require("lodash");
var utils_1 = require("../../../../utils");
var beforeAndAfter = function (size, variables, colors, props) { return (tslib_1.__assign(tslib_1.__assign({ content: '""', flex: 1 }, (props.vertical ? { width: size + 1 + "px", height: '100%' } : { height: size + 1 + "px" })), { background: _.get(colors, 'foreground', variables.dividerColor) })); };
exports.dividerStyles = {
    root: function (_a) {
        var props = _a.props, variables = _a.variables;
        var color = props.color, fitted = props.fitted, size = props.size, important = props.important, hasContent = props.hasContent, vertical = props.vertical;
        var colors = variables.colorScheme[color];
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ color: _.get(colors, 'foreground', variables.textColor), display: 'flex', alignItems: 'center' }, (!fitted && {
            padding: vertical ? "0 " + variables.dividerPadding : variables.dividerPadding + " 0",
        })), (important && {
            fontWeight: variables.importantFontWeight,
        })), (vertical && { height: '100%' })), (hasContent
            ? {
                textAlign: 'center',
                fontSize: utils_1.pxToRem(12 + size),
                lineHeight: variables.textLineHeight,
                '::before': tslib_1.__assign({}, beforeAndAfter(size, variables, colors, props)),
                '::after': tslib_1.__assign({}, beforeAndAfter(size, variables, colors, props)),
            }
            : {
                '::before': tslib_1.__assign({}, beforeAndAfter(size, variables, colors, props)),
            }));
    },
};
