// reakit/es/Combobox/ComboboxListState.js
import { _ as _objectWithoutPropertiesLoose, a as _objectSpread2 } from '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import 'reakit-utils/misc';
import 'react';
import { useInitialValue } from 'reakit-utils/hooks';
import 'reakit-utils/dom';
import '../reverse-30eaa122.js';
import '../getCurrentId-5aa9849e.js';
import '../findEnabledItemById-8ddca752.js';
import '../Id/IdProvider.js';
import '../Id/IdState.js';
import { useCompositeState } from '../Composite/CompositeState.js';
import { u as useComboboxBaseState } from '../ComboboxBaseState-73fabcba.js';

function unstable_useComboboxListState(initialState) {
  if (initialState === void 0) {
    initialState = {};
  }

  var _useInitialValue = useInitialValue(initialState),
      _useInitialValue$curr = _useInitialValue.currentId,
      currentId = _useInitialValue$curr === void 0 ? null : _useInitialValue$curr,
      _useInitialValue$orie = _useInitialValue.orientation,
      orientation = _useInitialValue$orie === void 0 ? "vertical" : _useInitialValue$orie,
      _useInitialValue$loop = _useInitialValue.loop,
      loop = _useInitialValue$loop === void 0 ? true : _useInitialValue$loop,
      sealed = _objectWithoutPropertiesLoose(_useInitialValue, ["currentId", "orientation", "loop"]);

  var composite = useCompositeState(_objectSpread2(_objectSpread2({
    currentId: currentId,
    orientation: orientation,
    loop: loop
  }, sealed), {}, {
    unstable_virtual: true,
    unstable_includesBaseElement: true
  }));
  return useComboboxBaseState(composite, sealed);
}

export { unstable_useComboboxListState };
