// reakit/es/Combobox/ComboboxGridCell.js
import '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import { createComponent } from 'reakit-system/createComponent';
import { createHook } from 'reakit-system/createHook';
import 'reakit-utils/misc';
import '../Box/Box.js';
import 'react';
import 'reakit-utils/hooks';
import 'reakit-utils/dom';
import 'reakit-warning';
import 'reakit-utils/events';
import 'reakit-utils/focus';
import '../Role/Role.js';
import '../Tabbable/Tabbable.js';
import '../Clickable/Clickable.js';
import '../getCurrentId-5aa9849e.js';
import '../__keys-6742f591.js';
import '../userFocus-e16425e3.js';
import { a as COMBOBOX_GRID_CELL_KEYS } from '../__keys-0f89298f.js';
import '../Id/IdProvider.js';
import '../Id/Id.js';
import '../setTextFieldValue-5c13f3b6.js';
import '../Composite/CompositeItem.js';
import '../__keys-08a69d36.js';
import { unstable_useGridCell } from '../Grid/GridCell.js';
import { unstable_useComboboxItem } from './ComboboxItem.js';

var unstable_useComboboxGridCell = createHook({
  name: "ComboboxGridCell",
  compose: [unstable_useComboboxItem, unstable_useGridCell],
  keys: COMBOBOX_GRID_CELL_KEYS
});
var unstable_ComboboxGridCell = createComponent({
  as: "span",
  memo: true,
  useHook: unstable_useComboboxGridCell
});

export { unstable_ComboboxGridCell, unstable_useComboboxGridCell };
