// reakit/es/Combobox/ComboboxOption.js
import { a as _objectSpread2 } from '../_rollupPluginBabelHelpers-1f0bf8c2.js';
import { createComponent } from 'reakit-system/createComponent';
import { createHook } from 'reakit-system/createHook';
import 'reakit-utils/misc';
import '../Box/Box.js';
import 'react';
import 'reakit-utils/hooks';
import 'reakit-utils/dom';
import 'reakit-warning';
import 'reakit-utils/events';
import 'reakit-utils/focus';
import '../Role/Role.js';
import '../Tabbable/Tabbable.js';
import '../Clickable/Clickable.js';
import '../getCurrentId-5aa9849e.js';
import '../__keys-6742f591.js';
import '../userFocus-e16425e3.js';
import { e as COMBOBOX_OPTION_KEYS } from '../__keys-0f89298f.js';
import '../Id/IdProvider.js';
import '../Id/Id.js';
import '../setTextFieldValue-5c13f3b6.js';
import { useCompositeItem } from '../Composite/CompositeItem.js';
import { unstable_useComboboxItem } from './ComboboxItem.js';

var unstable_useComboboxOption = createHook({
  name: "ComboboxOption",
  compose: [unstable_useComboboxItem, useCompositeItem],
  keys: COMBOBOX_OPTION_KEYS,
  useProps: function useProps(_, htmlProps) {
    return _objectSpread2({
      role: "option"
    }, htmlProps);
  }
});
var unstable_ComboboxOption = createComponent({
  as: "div",
  memo: true,
  useHook: unstable_useComboboxOption
});

export { unstable_ComboboxOption, unstable_useComboboxOption };
