// @fluentui/react-northstar/dist/commonjs/components/Input/InputLabel.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.InputLabel = exports.inputLabelClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _Box = require("../Box/Box");

var PropTypes = _interopRequireWildcard(require("prop-types"));

var inputLabelClassName = 'ui-input__label';
/**
 * An InputLabel provides a slot for label in the Input.
 */

exports.inputLabelClassName = inputLabelClassName;
var InputLabel = (0, _reactBindings.compose)(_Box.Box, {
  className: inputLabelClassName,
  displayName: 'InputLabel',
  overrideStyles: true,
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var labelPosition = _ref.labelPosition,
        required = _ref.required,
        hasValue = _ref.hasValue;
    return {
      labelPosition: labelPosition,
      required: required,
      hasValue: hasValue
    };
  },
  handledProps: ['required', 'labelPosition', 'hasValue'],
  shorthandConfig: {
    mappedProp: 'content'
  }
});
exports.InputLabel = InputLabel;
InputLabel.defaultProps = {
  as: 'label'
};
InputLabel.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  labelPosition: PropTypes.oneOf(['inline', 'above', 'inside']),
  required: PropTypes.bool,
  hasValue: PropTypes.bool
});
