// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Breadcrumb/breadcrumbItemStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.breadcrumbItemStyles = void 0;
var tslib_1 = require("tslib");
exports.breadcrumbItemStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ display: 'inline-flex', alignItems: 'center', verticalAlign: 'middle' }, (p.active && {
            fontWeight: v.itemCurrentFontWeight,
        })), (p.disabled && {
            color: v.disabledColor,
        })), (p.size === 'smaller' && {
            paddingLeft: v.linkPaddingLeftSmaller,
            paddingRight: v.linkPaddingRightSmaller,
            gap: v.linkSmallerGap,
        })), (p.size === 'small' && {
            paddingLeft: v.linkPaddingLeftSmall,
            paddingRight: v.linkPaddingRightSmall,
            gap: v.linkSmallGap,
        })), (p.size === 'medium' && {
            paddingLeft: v.linkPaddingLeftMedium,
            paddingRight: v.linkPaddingRightMedium,
            gap: v.linkMediumGap,
        })), (p.size === 'large' && {
            paddingLeft: v.linkPaddingLeftLarge,
            paddingRight: v.linkPaddingRightLarge,
            gap: v.linkLargeGap,
        })));
    },
};
