// @fluentui/react-northstar/dist/dts/src/components/Skeleton/SkeletonButton.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkeletonButton = exports.skeletonButtonClassName = void 0;
var tslib_1 = require("tslib");
var customPropTypes = require("@fluentui/react-proptypes");
var PropTypes = require("prop-types");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
var SkeletonShape_1 = require("./SkeletonShape");
exports.skeletonButtonClassName = 'ui-skeleton__button';
/**
 * An SkeletonButton represents a buton component that will be loaded
 */
exports.SkeletonButton = react_bindings_1.compose(SkeletonShape_1.SkeletonShape, {
    className: exports.skeletonButtonClassName,
    displayName: 'SkeletonButton',
    overrideStyles: true,
    shorthandConfig: {},
    handledProps: ['size', 'circular', 'iconOnly', 'fluid'],
    mapPropsToStylesProps: function (_a) {
        var size = _a.size, fluid = _a.fluid, iconOnly = _a.iconOnly, circular = _a.circular;
        return ({
            size: size,
            fluid: fluid,
            iconOnly: iconOnly,
            circular: circular,
        });
    },
});
exports.SkeletonButton.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { size: customPropTypes.size, circular: PropTypes.bool, iconOnly: PropTypes.bool, fluid: PropTypes.bool });
exports.SkeletonButton.defaultProps = {
    as: 'span',
    size: 'medium',
};
