// @fluentui/react-northstar/dist/dts/src/components/Skeleton/SkeletonInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkeletonInput = exports.skeletonInputClassName = void 0;
var tslib_1 = require("tslib");
var PropTypes = require("prop-types");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
var SkeletonShape_1 = require("./SkeletonShape");
exports.skeletonInputClassName = 'ui-skeleton__input';
/**
 * An SkeletonInput represents a input component that will be loaded
 */
exports.SkeletonInput = react_bindings_1.compose(SkeletonShape_1.SkeletonShape, {
    className: exports.skeletonInputClassName,
    displayName: 'SkeletonInput',
    overrideStyles: true,
    shorthandConfig: {},
    handledProps: ['fluid'],
    mapPropsToStylesProps: function (_a) {
        var fluid = _a.fluid;
        return ({
            fluid: fluid,
        });
    },
});
exports.SkeletonInput.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { fluid: PropTypes.bool });
exports.SkeletonInput.defaultProps = {
    as: 'span',
};
