// @fluentui/react-northstar/dist/dts/src/components/Skeleton/SkeletonLine.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkeletonLine = exports.skeletonLineClassName = void 0;
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var PropTypes = require("prop-types");
var utils_1 = require("../../utils");
var Box_1 = require("../Box/Box");
exports.skeletonLineClassName = 'ui-skeleton__line';
/**
 * A SkeletonLine represents a text line that will be loaded
 */
exports.SkeletonLine = react_bindings_1.compose(Box_1.Box, {
    className: exports.skeletonLineClassName,
    displayName: 'SkeletonLine',
    overrideStyles: true,
    shorthandConfig: {},
    handledProps: ['width', 'height'],
    mapPropsToStylesProps: function (_a) {
        var width = _a.width, height = _a.height;
        return ({ width: width, height: height });
    },
});
exports.SkeletonLine.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { width: PropTypes.string, height: PropTypes.string });
exports.SkeletonLine.defaultProps = {
    as: 'span',
    width: '100%',
    height: '1rem',
};
