// @fluentui/react-northstar/dist/dts/src/components/Skeleton/SkeletonAvatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkeletonAvatar = exports.skeletonAvatarClassName = void 0;
var tslib_1 = require("tslib");
var customPropTypes = require("@fluentui/react-proptypes");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
var SkeletonShape_1 = require("./SkeletonShape");
exports.skeletonAvatarClassName = 'ui-skeleton__avatar';
/**
 * An SkeletonAvatar represents an avatar component that will be loaded
 */
exports.SkeletonAvatar = react_bindings_1.compose(SkeletonShape_1.SkeletonShape, {
    className: exports.skeletonAvatarClassName,
    displayName: 'SkeletonAvatar',
    overrideStyles: true,
    shorthandConfig: {},
    handledProps: ['size'],
    mapPropsToStylesProps: function (_a) {
        var size = _a.size;
        return ({
            size: size,
        });
    },
});
exports.SkeletonAvatar.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { size: customPropTypes.size });
exports.SkeletonAvatar.defaultProps = {
    as: 'span',
    size: 'medium',
};
