// @fluentui/react-northstar/dist/commonjs/components/Skeleton/Skeleton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.Skeleton = exports.skeletonClassName = void 0;

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _Box = require("../Box/Box");

var _SkeletonLine = require("./SkeletonLine");

var _SkeletonShape = require("./SkeletonShape");

var _SkeletonButton = require("./SkeletonButton");

var _SkeletonText = require("./SkeletonText");

var _SkeletonInput = require("./SkeletonInput");

var _SkeletonAvatar = require("./SkeletonAvatar");

var skeletonClassName = 'ui-skeleton';
/**
 * A Skeleton is a component to be placed while the content is loading.
 */

exports.skeletonClassName = skeletonClassName;
var Skeleton = (0, _reactBindings.compose)(_Box.Box, {
  className: skeletonClassName,
  displayName: 'Skeleton',
  overrideStyles: true,
  shorthandConfig: {},
  handledProps: ['animation'],
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var animation = _ref.animation;
    return {
      animation: animation
    };
  }
});
exports.Skeleton = Skeleton;
Skeleton.Line = _SkeletonLine.SkeletonLine;
Skeleton.Shape = _SkeletonShape.SkeletonShape;
Skeleton.Button = _SkeletonButton.SkeletonButton;
Skeleton.Text = _SkeletonText.SkeletonText;
Skeleton.Input = _SkeletonInput.SkeletonInput;
Skeleton.Avatar = _SkeletonAvatar.SkeletonAvatar;
Skeleton.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon({
  accessibility: false
}), {
  animation: PropTypes.oneOf(['wave', 'pulse'])
});
