// @fluentui/react-northstar/dist/commonjs/components/Skeleton/SkeletonInput.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.SkeletonInput = exports.skeletonInputClassName = void 0;

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _SkeletonShape = require("./SkeletonShape");

var skeletonInputClassName = 'ui-skeleton__input';
/**
 * An SkeletonInput represents a input component that will be loaded
 */

exports.skeletonInputClassName = skeletonInputClassName;
var SkeletonInput = (0, _reactBindings.compose)(_SkeletonShape.SkeletonShape, {
  className: skeletonInputClassName,
  displayName: 'SkeletonInput',
  overrideStyles: true,
  shorthandConfig: {},
  handledProps: ['fluid'],
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var fluid = _ref.fluid;
    return {
      fluid: fluid
    };
  }
});
exports.SkeletonInput = SkeletonInput;
SkeletonInput.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  fluid: PropTypes.bool
});
SkeletonInput.defaultProps = {
  as: 'span'
};
