// @fluentui/react-northstar/dist/commonjs/components/Skeleton/SkeletonText.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.SkeletonText = exports.skeletonTextClassName = void 0;

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _SkeletonLine = require("./SkeletonLine");

var skeletonTextClassName = 'ui-skeleton__text';
/**
 * An SkeletonText represents a text component that will be loaded
 */

exports.skeletonTextClassName = skeletonTextClassName;
var SkeletonText = (0, _reactBindings.compose)(_SkeletonLine.SkeletonLine, {
  className: skeletonTextClassName,
  displayName: 'SkeletonText',
  overrideStyles: true,
  shorthandConfig: {},
  handledProps: ['size'],
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var size = _ref.size;
    return {
      size: size
    };
  }
});
exports.SkeletonText = SkeletonText;
SkeletonText.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  size: customPropTypes.size
});
SkeletonText.defaultProps = {
  as: 'span',
  size: 'medium'
};
