// @fluentui/react-northstar/dist/commonjs/components/Skeleton/SkeletonButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.SkeletonButton = exports.skeletonButtonClassName = void 0;

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _SkeletonShape = require("./SkeletonShape");

var skeletonButtonClassName = 'ui-skeleton__button';
/**
 * An SkeletonButton represents a buton component that will be loaded
 */

exports.skeletonButtonClassName = skeletonButtonClassName;
var SkeletonButton = (0, _reactBindings.compose)(_SkeletonShape.SkeletonShape, {
  className: skeletonButtonClassName,
  displayName: 'SkeletonButton',
  overrideStyles: true,
  shorthandConfig: {},
  handledProps: ['size', 'circular', 'iconOnly', 'fluid'],
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var size = _ref.size,
        fluid = _ref.fluid,
        iconOnly = _ref.iconOnly,
        circular = _ref.circular;
    return {
      size: size,
      fluid: fluid,
      iconOnly: iconOnly,
      circular: circular
    };
  }
});
exports.SkeletonButton = SkeletonButton;
SkeletonButton.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  size: customPropTypes.size,
  circular: PropTypes.bool,
  iconOnly: PropTypes.bool,
  fluid: PropTypes.bool
});
SkeletonButton.defaultProps = {
  as: 'span',
  size: 'medium'
};
