// @fluentui/react-northstar/dist/commonjs/components/Skeleton/SkeletonAvatar.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.SkeletonAvatar = exports.skeletonAvatarClassName = void 0;

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _SkeletonShape = require("./SkeletonShape");

var skeletonAvatarClassName = 'ui-skeleton__avatar';
/**
 * An SkeletonAvatar represents an avatar component that will be loaded
 */

exports.skeletonAvatarClassName = skeletonAvatarClassName;
var SkeletonAvatar = (0, _reactBindings.compose)(_SkeletonShape.SkeletonShape, {
  className: skeletonAvatarClassName,
  displayName: 'SkeletonAvatar',
  overrideStyles: true,
  shorthandConfig: {},
  handledProps: ['size'],
  mapPropsToStylesProps: function mapPropsToStylesProps(_ref) {
    var size = _ref.size;
    return {
      size: size
    };
  }
});
exports.SkeletonAvatar = SkeletonAvatar;
SkeletonAvatar.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  size: customPropTypes.size
});
SkeletonAvatar.defaultProps = {
  as: 'span',
  size: 'medium'
};
