// @fluentui/react-northstar/dist/dts/src/components/Pill/PillContent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PillContent = exports.pillContentClassName = void 0;
var tslib_1 = require("tslib");
var React = require("react");
var PropTypes = require("prop-types");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
exports.pillContentClassName = 'ui-pillcontent';
/**
 * A PillContent allows user to classify content.
 */
var PillContent = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.PillContent.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var accessibility = props.accessibility, children = props.children, className = props.className, content = props.content, design = props.design, styles = props.styles, variables = props.variables, size = props.size, actionable = props.actionable;
    var getA11Props = react_bindings_1.useAccessibility(accessibility, {
        debugName: exports.PillContent.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(exports.PillContent.displayName, {
        className: exports.pillContentClassName,
        mapPropsToStyles: function () { return ({ size: size, actionable: actionable }); },
        mapPropsToInlineStyles: function () { return ({ className: className, design: design, styles: styles, variables: variables }); },
        rtl: context.rtl,
    }).classes;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.PillContent.handledProps, props);
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign(tslib_1.__assign({ className: classes.root }, utils_1.rtlTextContainer.getAttributes({ forElements: [children] })), unhandledProps))), utils_1.childrenExist(children) ? children : content));
    setEnd();
    return element;
};
exports.PillContent = PillContent;
exports.PillContent.displayName = 'PillContent';
exports.PillContent.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { actionable: PropTypes.bool });
exports.PillContent.handledProps = Object.keys(exports.PillContent.propTypes);
exports.PillContent.defaultProps = {
    as: 'span',
};
exports.PillContent.shorthandConfig = {
    mappedProp: 'content',
};
