// @fluentui/react-northstar/dist/dts/src/components/Pill/PillAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PillAction = exports.pillActionClassName = void 0;
var tslib_1 = require("tslib");
var React = require("react");
var accessibility_1 = require("@fluentui/accessibility");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
var react_icons_northstar_1 = require("@fluentui/react-icons-northstar");
exports.pillActionClassName = 'ui-pill__action';
/**
 * A PillAction allows user to execute an action.
 */
var PillAction = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.PillAction.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var accessibility = props.accessibility, children = props.children, className = props.className, content = props.content, design = props.design, styles = props.styles, variables = props.variables, size = props.size;
    var getA11Props = react_bindings_1.useAccessibility(accessibility, {
        debugName: exports.PillAction.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(exports.PillAction.displayName, {
        className: exports.pillActionClassName,
        mapPropsToStyles: function () { return ({ size: size }); },
        mapPropsToInlineStyles: function () { return ({ className: className, design: design, styles: styles, variables: variables }); },
        rtl: context.rtl,
    }).classes;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.PillAction.handledProps, props);
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign(tslib_1.__assign({ className: classes.root }, utils_1.rtlTextContainer.getAttributes({ forElements: [children] })), unhandledProps))), utils_1.childrenExist(children) ? children : content));
    setEnd();
    return element;
};
exports.PillAction = PillAction;
exports.PillAction.displayName = 'PillAction';
exports.PillAction.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon());
exports.PillAction.handledProps = Object.keys(exports.PillAction.propTypes);
exports.PillAction.defaultProps = {
    accessibility: accessibility_1.pillActionBehavior,
    content: React.createElement(react_icons_northstar_1.CloseIcon, null),
};
exports.PillAction.shorthandConfig = {
    mappedProp: 'content',
};
