// @fluentui/react-northstar/dist/dts/src/components/Pill/PillImage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PillImage = exports.pillImageClassName = void 0;
var tslib_1 = require("tslib");
var accessibility_1 = require("@fluentui/accessibility");
var react_bindings_1 = require("@fluentui/react-bindings");
var React = require("react");
var utils_1 = require("../../utils");
exports.pillImageClassName = 'ui-pill__image';
/**
 * An PillImage is a graphic representation used by Pill.
 */
var PillImage = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.PillImage.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var accessibility = props.accessibility, className = props.className, design = props.design, styles = props.styles, variables = props.variables, size = props.size;
    var getA11Props = react_bindings_1.useAccessibility(accessibility, {
        debugName: exports.PillImage.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(exports.PillImage.displayName, {
        className: exports.pillImageClassName,
        mapPropsToStyles: function () { return ({
            size: size,
        }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.PillImage.handledProps, props);
    var result = React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign({ className: classes.root }, unhandledProps))));
    setEnd();
    return result;
};
exports.PillImage = PillImage;
exports.PillImage.displayName = 'PillImage';
exports.PillImage.defaultProps = {
    accessibility: accessibility_1.pillImageBehavior,
    as: 'img',
};
exports.PillImage.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({
    children: false,
    content: false,
}));
exports.PillImage.handledProps = Object.keys(exports.PillImage.propTypes);
exports.PillImage.shorthandConfig = {
    mappedProp: 'src',
    allowsJSX: false,
};
