// @fluentui/accessibility/dist/dts/src/behaviors/Button/buttonBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonBehavior = void 0;
var tslib_1 = require("tslib");
var keyboard_key_1 = require("../../keyboard-key");
var buttonBehavior = function (props) {
    var definition = {
        attributes: {
            root: {
                role: props.as === 'button' ? undefined : 'button',
                tabIndex: props.as === 'button' || props.disabled ? undefined : 0,
                disabled: props.as === 'button' ? props.disabled : undefined,
                'aria-disabled': props.disabledFocusable,
            },
        },
        keyActions: {
            root: tslib_1.__assign({}, (props.as !== 'button' &&
                props.as !== 'a' && {
                performClick: {
                    keyCombinations: [{ keyCode: keyboard_key_1.keyboardKey.Enter }, { keyCode: keyboard_key_1.SpacebarKey }],
                },
            })),
        },
    };
    if (process.env.NODE_ENV !== 'production') {
        // Override the default trigger's accessibility schema class.
        if (props.disabledFocusable) {
            definition.attributes.root['data-aa-class'] = 'DisabledFocusableButton';
        }
    }
    return definition;
};
exports.buttonBehavior = buttonBehavior;
