// @fluentui/react-bindings/dist/dts/test/compose/mergeProps-test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var consts_1 = require("../../src/compose/consts");
describe('mergeProps', function () {
    var selfSlot = { __self: consts_1.defaultComposeOptions.slots.__self };
    it('can pass through default options', function () {
        expect(react_bindings_1.mergeProps({}, consts_1.defaultComposeOptions)).toEqual({
            state: {},
            slots: tslib_1.__assign(tslib_1.__assign({}, selfSlot), { root: 'div' }),
            slotProps: {
                root: {},
            },
        });
    });
    it('can pass through default slot props', function () {
        expect(react_bindings_1.mergeProps({ id: 'abc' }, tslib_1.__assign(tslib_1.__assign({}, consts_1.defaultComposeOptions), { handledProps: ['id'], slotProps: [
                function (_state) {
                    return {
                        slot1: { id: 'not this one' },
                    };
                },
                function (state) {
                    return {
                        slot1: { id: state.id },
                    };
                },
            ] }))).toEqual({
            state: { id: 'abc' },
            slots: tslib_1.__assign(tslib_1.__assign({}, selfSlot), { root: 'div' }),
            slotProps: {
                root: {},
                slot1: { id: 'abc' },
            },
        });
    });
    it('can resolve "as" prop', function () {
        var state = { as: 'button' };
        expect(react_bindings_1.mergeProps(state, consts_1.defaultComposeOptions)).toEqual({
            state: state,
            slots: tslib_1.__assign(tslib_1.__assign({}, selfSlot), { root: 'button' }),
            slotProps: {
                root: {},
            },
        });
    });
    it('can mix unrecognized props onto the root', function () {
        expect(react_bindings_1.mergeProps({ 'data-foo': 'foo' }, consts_1.defaultComposeOptions)).toEqual({
            slots: tslib_1.__assign(tslib_1.__assign({}, selfSlot), { root: 'div' }),
            state: {
                'data-foo': 'foo',
            },
            slotProps: {
                root: {
                    'data-foo': 'foo',
                },
            },
        });
    });
});
