// react-cosmos/dist/dom/DomFixtureLoader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DomFixtureLoader = DomFixtureLoader;

var _react = _interopRequireDefault(require("react"));

var _FixtureLoader = require("react-cosmos-shared2/FixtureLoader");

var _ErrorCatch = require("./ErrorCatch");

var _rendererConnect = require("./rendererConnect");

var _rendererId = require("./rendererId");

var _renderMessage = require("./renderMessage");

var _selectedFixtureId = require("./selectedFixtureId");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function DomFixtureLoader(_ref) {
  var fixtures = _ref.fixtures,
      decorators = _ref.decorators,
      onErrorReset = _ref.onErrorReset;
  return /*#__PURE__*/_react.default.createElement(_FixtureLoader.FixtureLoader, {
    rendererId: _rendererId.rendererId,
    rendererConnect: _rendererConnect.rendererConnect,
    fixtures: fixtures,
    selectedFixtureId: _selectedFixtureId.selectedFixtureId,
    systemDecorators: [_ErrorCatch.ErrorCatch],
    userDecorators: decorators,
    renderMessage: _renderMessage.renderMessage,
    onErrorReset: onErrorReset
  });
}
