// @fluentui/react-northstar/dist/commonjs/components/Design/Design.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.Design = Design;

var _reactBindings = require("@fluentui/react-bindings");

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var React = _interopRequireWildcard(require("react"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

/**
 * The Design component provides a theme safe subset of CSS for designing layouts.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
function Design(_ref) {
  var config = _ref.config,
      children = _ref.children;
  var context = (0, _reactBindings.useFluentContext)();
  var getConfig = React.useCallback(function () {
    return config;
  }, [config]); // Heads Up! Keep in sync with renderComponent.tsx

  var styleParam = {
    displayName: Design.displayName,
    disableAnimations: context.disableAnimations,
    direction: context.rtl ? 'rtl' : 'ltr',
    sanitizeCss: context.performance.enableSanitizeCssPlugin
  };
  var className = context.renderer.renderRule(getConfig, styleParam);
  return children({
    className: className
  });
}

Design.displayName = 'Design';
Design.propTypes = {
  children: PropTypes.func.isRequired,
  config: customPropTypes.design.isRequired
};
