// @fluentui/accessibility/dist/commonjs/behaviors/Tab/tabBehavior.js
"use strict";

exports.__esModule = true;
exports.tabBehavior = void 0;

var _keyboardKey = require("../../keyboard-key");

/**
 * @specification
 * Adds role 'presentation' to 'wrapper' slot.
 * Adds role 'tab' to 'root' slot.
 * Adds attribute 'tabIndex=0' to 'root' slot.
 * Adds attribute 'aria-selected=true' based on the property 'active'. This can be overriden by providing 'aria-selected' property directly to the component.
 * Adds attribute 'aria-label' based on the property 'aria-label' to 'root' slot.
 * Adds attribute 'aria-labelledby' based on the property 'aria-labelledby' to 'root' slot.
 * Adds attribute 'aria-describedby' based on the property 'aria-describedby' to 'root' slot.
 * Adds attribute 'aria-controls' based on the property 'aria-controls' to 'root' slot.
 * Adds attribute 'aria-disabled=true' based on the property 'disabled'. This can be overriden by providing 'aria-disabled' property directly to the component.
 * Triggers 'performClick' action with 'Enter' or 'Spacebar' on 'root'.
 */
var tabBehavior = function tabBehavior(props) {
  return {
    attributes: {
      wrapper: {
        role: 'presentation'
      },
      root: {
        role: 'tab',
        tabIndex: 0,
        'aria-selected': !!props.active,
        'aria-label': props['aria-label'],
        'aria-labelledby': props['aria-labelledby'],
        'aria-describedby': props['aria-describedby'],
        'aria-controls': props['aria-controls'],
        'aria-disabled': props['disabled']
      }
    },
    keyActions: {
      root: {
        performClick: {
          keyCombinations: [{
            keyCode: _keyboardKey.keyboardKey.Enter
          }, {
            keyCode: _keyboardKey.SpacebarKey
          }]
        }
      }
    }
  };
};

exports.tabBehavior = tabBehavior;
