// @fluentui/react-northstar-fela-renderer/dist/dts/src/felaExpandCssShorthandsPlugin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.felaExpandCssShorthandsPlugin = void 0;
var tslib_1 = require("tslib");
// @ts-ignore
var inline_style_expand_shorthand_1 = require("inline-style-expand-shorthand");
// https://jsperf.com/array-indexof-vs-object-key-lookup2/12
var handledCssProps = {
    // 'font', Oops, is not supported by inline-style-expand-shorthand
    padding: true,
    margin: true,
    border: true,
    borderWidth: true,
    borderStyle: true,
    borderColor: true,
    borderTop: true,
    borderRight: true,
    borderBottom: true,
    borderLeft: true,
    borderRadius: true,
    background: true,
    outline: true,
};
var felaExpandCssShorthandsPlugin = function (styles) {
    return Object.keys(styles).reduce(function (acc, cssPropertyName) {
        var _a, _b, _c, _d;
        var cssPropertyValue = styles[cssPropertyName];
        if (cssPropertyValue === null || typeof cssPropertyValue === 'undefined') {
            return tslib_1.__assign(tslib_1.__assign({}, acc), (_a = {}, _a[cssPropertyName] = cssPropertyValue, _a));
        }
        if (handledCssProps[cssPropertyName]) {
            var expandedProps = inline_style_expand_shorthand_1.expandProperty(cssPropertyName, cssPropertyValue);
            if (expandedProps) {
                return tslib_1.__assign(tslib_1.__assign({}, acc), expandedProps);
            }
        }
        if (Array.isArray(cssPropertyValue)) {
            return tslib_1.__assign(tslib_1.__assign({}, acc), (_b = {}, _b[cssPropertyName] = cssPropertyValue, _b));
        }
        if (typeof cssPropertyValue === 'object') {
            return tslib_1.__assign(tslib_1.__assign({}, acc), (_c = {}, _c[cssPropertyName] = exports.felaExpandCssShorthandsPlugin(cssPropertyValue), _c));
        }
        return tslib_1.__assign(tslib_1.__assign({}, acc), (_d = {}, _d[cssPropertyName] = cssPropertyValue, _d));
    }, {});
};
exports.felaExpandCssShorthandsPlugin = felaExpandCssShorthandsPlugin;
