// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Table/tableCellStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tableCellStyles = void 0;
var tslib_1 = require("tslib");
var getBorderFocusStyles_1 = require("../../getBorderFocusStyles");
exports.tableCellStyles = {
    root: function (_a) {
        var v = _a.variables, siteVariables = _a.theme.siteVariables;
        var borderFocusStyles = getBorderFocusStyles_1.getBorderFocusStyles({
            variables: siteVariables,
        });
        return tslib_1.__assign(tslib_1.__assign({ display: 'flex', flexFlow: 'row nowrap', flexGrow: 1, flexBasis: 0, minWidth: v.minCellWidth, outline: 0, borderWidth: v.borderWidth, borderStyle: 'solid', borderColor: 'transparent' }, borderFocusStyles), { padding: v.cellPadding, position: 'relative', height: '100%' });
    },
    content: function (_a) {
        var p = _a.props;
        return tslib_1.__assign({ alignSelf: 'center' }, (p.truncateContent && {
            display: 'block',
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            whiteSpace: 'nowrap',
        }));
    },
};
