// @fluentui/react-northstar/dist/dts/src/components/Dialog/DialogFooter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DialogFooter = exports.dialogFooterClassName = void 0;
var tslib_1 = require("tslib");
var React = require("react");
var utils_1 = require("../../utils");
var react_bindings_1 = require("@fluentui/react-bindings");
exports.dialogFooterClassName = 'ui-dialog__footer';
/**
 * A DialogFooter represents footer content in Dialog, usually shows dialog actions.
 */
var DialogFooter = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.DialogFooter.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var children = props.children, content = props.content, className = props.className, design = props.design, styles = props.styles, variables = props.variables, accessibility = props.accessibility;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.DialogFooter.handledProps, props);
    var getA11yProps = react_bindings_1.useAccessibility(accessibility, {
        debugName: exports.DialogFooter.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(exports.DialogFooter.displayName, {
        className: exports.dialogFooterClassName,
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: variables,
        }); },
        rtl: context.rtl,
    }).classes;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign({ className: classes.root }, unhandledProps))), utils_1.childrenExist(children) ? children : content));
    setEnd();
    return element;
};
exports.DialogFooter = DialogFooter;
exports.DialogFooter.displayName = 'DialogFooter';
exports.DialogFooter.propTypes = tslib_1.__assign({}, utils_1.commonPropTypes.createCommon());
exports.DialogFooter.handledProps = Object.keys(exports.DialogFooter.propTypes);
exports.DialogFooter.create = utils_1.createShorthandFactory({ Component: exports.DialogFooter, mappedProp: 'content' });
