// @fluentui/react-northstar/dist/dts/src/components/Label/Label.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Label = exports.labelClassName = void 0;
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var customPropTypes = require("@fluentui/react-proptypes");
var _ = require("lodash");
var PropTypes = require("prop-types");
var React = require("react");
var utils_1 = require("../../utils");
var Image_1 = require("../Image/Image");
var Box_1 = require("../Box/Box");
exports.labelClassName = 'ui-label';
/**
 * A Label allows user to classify content.
 */
var Label = function (props) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(exports.Label.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var accessibility = props.accessibility, children = props.children, className = props.className, circular = props.circular, color = props.color, content = props.content, icon = props.icon, iconPosition = props.iconPosition, design = props.design, styles = props.styles, variables = props.variables, image = props.image, imagePosition = props.imagePosition;
    var getA11Props = react_bindings_1.useAccessibility(accessibility, {
        debugName: exports.Label.displayName,
        rtl: context.rtl,
    });
    var _b = react_bindings_1.useStyles(exports.Label.displayName, {
        className: exports.labelClassName,
        mapPropsToStyles: function () { return ({
            hasActionableIcon: _.has(icon, 'onClick'),
            hasImage: !!image,
            hasIcon: !!icon,
            circular: circular,
            color: color,
            imagePosition: imagePosition,
            iconPosition: iconPosition,
        }); },
        mapPropsToInlineStyles: function () { return ({ className: className, design: design, styles: styles, variables: variables }); },
        rtl: context.rtl,
    }), classes = _b.classes, resolvedStyles = _b.styles;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(exports.Label.handledProps, props);
    if (utils_1.childrenExist(children)) {
        var element_1 = (React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign(tslib_1.__assign({ className: classes.root }, utils_1.rtlTextContainer.getAttributes({ forElements: [children] })), unhandledProps))), children));
        setEnd();
        return element_1;
    }
    var imageElement = Image_1.Image.create(image, {
        defaultProps: function () { return ({
            styles: resolvedStyles.image,
        }); },
    });
    var iconElement = Box_1.Box.create(icon, {
        defaultProps: function () { return ({
            styles: resolvedStyles.icon,
        }); },
    });
    var contentElement = Box_1.Box.create(content, {
        defaultProps: function () { return ({
            styles: resolvedStyles.content,
        }); },
    });
    var startImage = imagePosition === 'start' && imageElement;
    var startIcon = iconPosition === 'start' && iconElement;
    var endIcon = iconPosition === 'end' && iconElement;
    var endImage = imagePosition === 'end' && imageElement;
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11Props('root', tslib_1.__assign({ className: classes.root }, unhandledProps))),
        startImage,
        startIcon,
        contentElement,
        endIcon,
        endImage));
    setEnd();
    return element;
};
exports.Label = Label;
exports.Label.displayName = 'Label';
exports.Label.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({ color: true, content: 'shorthand' })), { circular: PropTypes.bool, icon: customPropTypes.shorthandAllowingChildren, iconPosition: PropTypes.oneOf(['start', 'end']), image: customPropTypes.itemShorthandWithoutJSX, imagePosition: PropTypes.oneOf(['start', 'end']), fluid: PropTypes.bool });
exports.Label.handledProps = Object.keys(exports.Label.propTypes);
exports.Label.defaultProps = {
    as: 'span',
    imagePosition: 'start',
    iconPosition: 'end',
};
exports.Label.create = utils_1.createShorthandFactory({ Component: exports.Label, mappedProp: 'content' });
