// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Tooltip/tooltipContentStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tooltipContentStyles = void 0;
var tslib_1 = require("tslib");
var getPointerStyles_1 = require("../../getPointerStyles");
var pointerSvgUrl_1 = require("../../pointerSvgUrl");
exports.tooltipContentStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign({ display: 'none', position: 'absolute', maxWidth: v.maxWidth, zIndex: v.zIndex }, (p.pointing && tslib_1.__assign({ pointerEvents: 'all' }, getPointerStyles_1.getContainerStyles({
            placement: p.basePlacement,
            padding: v.pointerMargin,
        })))), (p.open && {
            display: 'block',
        })));
    },
    pointer: function (_a) {
        var p = _a.props, v = _a.variables, rtl = _a.rtl;
        return (tslib_1.__assign({ display: 'block', position: 'absolute', width: v.pointerWidth, height: v.pointerHeight }, getPointerStyles_1.getPointerStyles({
            backgroundColor: v.backgroundColor,
            borderSize: v.borderSize,
            borderColor: 'transparent',
            gap: v.pointerGap,
            padding: v.pointerMargin,
            height: v.pointerHeight,
            width: v.pointerWidth,
            placement: p.basePlacement,
            rtl: rtl,
            svg: pointerSvgUrl_1.pointerSvgUrl(v.backgroundColor),
        })));
    },
    content: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign({ display: 'block', padding: v.padding, textAlign: 'left', color: v.color, background: v.backgroundColor, borderRadius: v.borderRadius, boxShadow: v.boxShadow }, (p.pointing && {
            pointerEvents: 'all',
        })));
    },
};
