// @fluentui/react-bindings/dist/commonjs/context.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.useFluentContext = useFluentContext;
exports.Unstable_FluentContextProvider = exports.defaultContextValue = exports.defaultPerformanceFlags = void 0;

var _reactNorthstarStylesRenderer = require("@fluentui/react-northstar-styles-renderer");

var _styles = require("@fluentui/styles");

var React = _interopRequireWildcard(require("react"));

var defaultPerformanceFlags = {
  enableSanitizeCssPlugin: process.env.NODE_ENV !== 'production',
  enableStylesCaching: true,
  enableVariablesCaching: true,
  enableBooleanVariablesCaching: false
};
exports.defaultPerformanceFlags = defaultPerformanceFlags;
var defaultContextValue = {
  // A default value for `rtl` is undefined to let compute `Provider` a proper one
  rtl: undefined,
  disableAnimations: false,
  performance: defaultPerformanceFlags,
  renderer: _reactNorthstarStylesRenderer.noopRenderer,
  theme: _styles.emptyTheme,
  telemetry: undefined,
  target: undefined
};
exports.defaultContextValue = defaultContextValue;
var FluentContext = /*#__PURE__*/React.createContext(defaultContextValue);

function useFluentContext() {
  return React.useContext(FluentContext);
}

var Unstable_FluentContextProvider = FluentContext.Provider;
exports.Unstable_FluentContextProvider = Unstable_FluentContextProvider;
