// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Accordion/accordionTitleStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.accordionTitleStyles = void 0;
var tslib_1 = require("tslib");
var activeIndicatorUrl_1 = require("./activeIndicatorUrl");
var utils_1 = require("../../../../utils");
exports.accordionTitleStyles = {
    root: function (_a) {
        var p = _a.props;
        return ({
            display: 'inline-block',
            verticalAlign: 'middle',
            padding: '.5rem 0',
            cursor: p.disabled ? 'default' : 'pointer',
        });
    },
    contentWrapper: function (_a) {
        var p = _a.props;
        return (tslib_1.__assign({ display: 'grid', gridTemplateColumns: 'auto', msGridColumns: 'auto', userSelect: 'none' }, (p.content && {
            gridTemplateColumns: 'auto 1fr',
            msGridColumns: 'auto 1fr',
        })));
    },
    indicator: function (_a) {
        var p = _a.props, v = _a.variables, rtl = _a.rtl;
        return (tslib_1.__assign(tslib_1.__assign({ alignItems: 'center', display: 'inline-flex', backgroundImage: activeIndicatorUrl_1.activeIndicatorUrl(v.color, p.active), backgroundPosition: 'center' }, (rtl && {
            transform: "scaleX(-1)",
        })), { content: '" "', overflow: 'hidden', height: '100%', width: utils_1.pxToRem(16), msGridColumn: '1', marginRight: "" + utils_1.pxToRem(4) }));
    },
    content: function () { return ({
        alignItems: 'center',
        display: 'grid',
        msGridColumn: '2',
    }); },
};
