// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Button/buttonGroupStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonGroupStyles = void 0;
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
var commonButtonsStyles = function (circular) { return (circular ? { marginRight: utils_1.pxToRem(8) } : { borderRadius: 0 }); };
exports.buttonGroupStyles = {
    root: function () { return ({}); },
    middleButton: function (_a) {
        var p = _a.props;
        return (tslib_1.__assign({}, commonButtonsStyles(p.circular)));
    },
    firstButton: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign({}, commonButtonsStyles(p.circular)), (!p.circular && {
            borderTopLeftRadius: v.borderRadius,
            borderBottomLeftRadius: v.borderRadius,
        })));
    },
    lastButton: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign({}, commonButtonsStyles(p.circular)), (!p.circular && {
            borderTopRightRadius: v.borderRadius,
            borderBottomRightRadius: v.borderRadius,
        })));
    },
};
