// @fluentui/react-northstar/dist/dts/src/components/Portal/PortalInner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PortalInner = void 0;
var tslib_1 = require("tslib");
var _ = require("lodash");
var PropTypes = require("prop-types");
var React = require("react");
var ReactDOM = require("react-dom");
var utils_1 = require("../../utils");
var usePortalBox_1 = require("../Provider/usePortalBox");
var customPropTypes = require("@fluentui/react-proptypes");
var react_bindings_1 = require("@fluentui/react-bindings");
/**
 * A PortalInner is a container for Portal's content.
 */
var PortalInner = function (props) {
    var context = React.useContext(usePortalBox_1.PortalBoxContext);
    var children = props.children, mountNode = props.mountNode;
    // PortalInner should render elements even without a context
    // eslint-disable-next-line
    var target = utils_1.isBrowser() ? context || document.body : null;
    var container = mountNode || target;
    react_bindings_1.useIsomorphicLayoutEffect(function () {
        _.invoke(props, 'onMount', props);
        return function () { return _.invoke(props, 'onUnmount', props); };
    }, []);
    return container && ReactDOM.createPortal(children, container);
};
exports.PortalInner = PortalInner;
exports.PortalInner.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon({
    accessibility: false,
    as: false,
    className: false,
    content: false,
    styled: false,
})), { mountNode: customPropTypes.domNode, onMount: PropTypes.func, onUnmount: PropTypes.func });
