// reakit-utils/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var misc = require('./misc.js');
var dom = require('./dom.js');
require('./_rollupPluginBabelHelpers-0184dad2.js');
var events = require('./events.js');
var flatten = require('./flatten.js');
var focus = require('./focus.js');
require('react');
var hooks = require('./hooks.js');



exports.applyState = misc.applyState;
exports.isEmpty = misc.isEmpty;
exports.isInteger = misc.isInteger;
exports.isObject = misc.isObject;
exports.isPlainObject = misc.isPlainObject;
exports.isPromise = misc.isPromise;
exports.omit = misc.omit;
exports.pick = misc.pick;
exports.removeIndexFromArray = misc.removeIndexFromArray;
exports.removeItemFromArray = misc.removeItemFromArray;
exports.shallowEqual = misc.shallowEqual;
exports.toArray = misc.toArray;
exports.canUseDOM = dom.canUseDOM;
exports.closest = dom.closest;
exports.contains = dom.contains;
exports.getActiveElement = dom.getActiveElement;
exports.getDocument = dom.getDocument;
exports.getNativeElementType = dom.getNativeElementType;
exports.getWindow = dom.getWindow;
exports.isButton = dom.isButton;
exports.isTextField = dom.isTextField;
exports.isUA = dom.isUA;
exports.matches = dom.matches;
exports.createEvent = events.createEvent;
exports.fireBlurEvent = events.fireBlurEvent;
exports.fireEvent = events.fireEvent;
exports.fireKeyboardEvent = events.fireKeyboardEvent;
exports.getNextActiveElementOnBlur = events.getNextActiveElementOnBlur;
exports.isPortalEvent = events.isPortalEvent;
exports.isSelfTarget = events.isSelfTarget;
exports.flatten = flatten.flatten;
exports.ensureFocus = focus.ensureFocus;
exports.getAllFocusableIn = focus.getAllFocusableIn;
exports.getAllTabbableIn = focus.getAllTabbableIn;
exports.getClosestFocusable = focus.getClosestFocusable;
exports.getFirstFocusableIn = focus.getFirstFocusableIn;
exports.getFirstTabbableIn = focus.getFirstTabbableIn;
exports.getLastTabbableIn = focus.getLastTabbableIn;
exports.getNextTabbableIn = focus.getNextTabbableIn;
exports.getPreviousTabbableIn = focus.getPreviousTabbableIn;
exports.hasFocus = focus.hasFocus;
exports.hasFocusWithin = focus.hasFocusWithin;
exports.isFocusable = focus.isFocusable;
exports.isTabbable = focus.isTabbable;
exports.useEventCallback = hooks.useEventCallback;
exports.useForkRef = hooks.useForkRef;
exports.useInitialValue = hooks.useInitialValue;
exports.useLazyRef = hooks.useLazyRef;
exports.useLiveRef = hooks.useLiveRef;
exports.useNativeElementType = hooks.useNativeElementType;
exports.useSafeLayoutEffect = hooks.useSafeLayoutEffect;
exports.useUpdateEffect = hooks.useUpdateEffect;
