// @fluentui/react-northstar/dist/commonjs/components/Pill/PillAction.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.PillAction = exports.pillActionClassName = void 0;

var React = _interopRequireWildcard(require("react"));

var _accessibility = require("@fluentui/accessibility");

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _reactIconsNorthstar = require("@fluentui/react-icons-northstar");

var pillActionClassName = 'ui-pill__action';
/**
 * A PillAction allows user to execute an action.
 */

exports.pillActionClassName = pillActionClassName;

var PillAction = function PillAction(props) {
  var context = (0, _reactBindings.useFluentContext)();

  var _useTelemetry = (0, _reactBindings.useTelemetry)(PillAction.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var accessibility = props.accessibility,
      children = props.children,
      className = props.className,
      content = props.content,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      size = props.size;
  var getA11Props = (0, _reactBindings.useAccessibility)(accessibility, {
    debugName: PillAction.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(PillAction.displayName, {
    className: pillActionClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        size: size
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(PillAction.handledProps, props);
  var element = /*#__PURE__*/React.createElement(ElementType, getA11Props('root', Object.assign({
    className: classes.root
  }, _utils.rtlTextContainer.getAttributes({
    forElements: [children]
  }), unhandledProps)), (0, _utils.childrenExist)(children) ? children : content);
  setEnd();
  return element;
};

exports.PillAction = PillAction;
PillAction.displayName = 'PillAction';
PillAction.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon());
PillAction.handledProps = Object.keys(PillAction.propTypes);
PillAction.defaultProps = {
  accessibility: _accessibility.pillActionBehavior,
  content: /*#__PURE__*/React.createElement(_reactIconsNorthstar.CloseIcon, null)
};
PillAction.shorthandConfig = {
  mappedProp: 'content'
};
