// @fluentui/react-northstar/dist/dts/src/themes/teams/components/TextArea/textAreaStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.textAreaStyles = void 0;
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
exports.textAreaStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ margin: v.margin, height: v.height, backgroundColor: v.backgroundColor }, (p.inverted && {
            backgroundColor: v.invertedBackgroundColor,
        })), { color: v.fontColor, borderColor: v.borderColor, borderRadius: v.borderRadius, borderStyle: 'solid', borderWidth: v.borderWidth, outline: 0, padding: v.padding, resize: p.resize || 'none' }), (p.fluid && {
            width: '100%',
        })), (p.disabled && {
            pointerEvents: 'none',
            color: v.disabledColor,
            boxShadow: 'none',
        })), (p.error && { border: utils_1.pxToRem(1) + " solid " + v.borderColorError })), { '::placeholder': tslib_1.__assign({ color: v.placeholderColor, opacity: 1 }, (p.disabled && {
                color: v.disabledColor,
            })), ':focus': {
                borderColor: v.borderColorFocus,
            } }));
    },
};
