// @fluentui/react-northstar/dist/dts/test/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mountWithProviderAndGetComponent = exports.mountWithProvider = exports.EmptyThemeProvider = exports.nextFrame = exports.getDisplayName = exports.findIntrinsicElement = exports.syntheticEvent = exports.domEvent = exports.consoleUtil = void 0;
var tslib_1 = require("tslib");
var consoleUtil = require("./consoleUtil");
exports.consoleUtil = consoleUtil;
var domEvent = require("./domEvent");
exports.domEvent = domEvent;
var syntheticEvent = require("./syntheticEvent");
exports.syntheticEvent = syntheticEvent;
tslib_1.__exportStar(require("./assertNodeContains"), exports);
var findIntrinsicElement_1 = require("./findIntrinsicElement");
Object.defineProperty(exports, "findIntrinsicElement", { enumerable: true, get: function () { return findIntrinsicElement_1.findIntrinsicElement; } });
var getDisplayName_1 = require("./getDisplayName");
Object.defineProperty(exports, "getDisplayName", { enumerable: true, get: function () { return getDisplayName_1.getDisplayName; } });
var nextFrame_1 = require("./nextFrame");
Object.defineProperty(exports, "nextFrame", { enumerable: true, get: function () { return nextFrame_1.nextFrame; } });
var withProvider_1 = require("./withProvider");
Object.defineProperty(exports, "EmptyThemeProvider", { enumerable: true, get: function () { return withProvider_1.EmptyThemeProvider; } });
Object.defineProperty(exports, "mountWithProvider", { enumerable: true, get: function () { return withProvider_1.mountWithProvider; } });
Object.defineProperty(exports, "mountWithProviderAndGetComponent", { enumerable: true, get: function () { return withProvider_1.mountWithProviderAndGetComponent; } });
