// @fluentui/react-northstar/dist/dts/test/utils/withProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mountWithProviderAndGetComponent = exports.mountWithProvider = exports.EmptyThemeProvider = void 0;
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var react_northstar_styles_renderer_1 = require("@fluentui/react-northstar-styles-renderer");
var styles_1 = require("@fluentui/styles");
var enzyme_1 = require("enzyme");
var React = require("react");
var EmptyThemeProvider = function (_a) {
    var children = _a.children, disableAnimations = _a.disableAnimations, _b = _a.renderer, renderer = _b === void 0 ? react_northstar_styles_renderer_1.noopRenderer : _b, telemetry = _a.telemetry, _c = _a.theme, theme = _c === void 0 ? styles_1.emptyTheme : _c, _d = _a.rtl, rtl = _d === void 0 ? false : _d;
    var value = {
        renderer: renderer,
        target: document,
        disableAnimations: disableAnimations,
        rtl: rtl,
        theme: theme,
        telemetry: telemetry,
        performance: {},
    };
    return React.createElement(react_bindings_1.Unstable_FluentContextProvider, { value: value }, children);
};
exports.EmptyThemeProvider = EmptyThemeProvider;
// eslint-disable-next-line @typescript-eslint/no-unused-vars
var mountWithProvider = function (node, options) {
    return enzyme_1.mount(node, tslib_1.__assign({ wrappingComponent: exports.EmptyThemeProvider }, options));
};
exports.mountWithProvider = mountWithProvider;
// eslint-disable-next-line @typescript-eslint/no-unused-vars
var mountWithProviderAndGetComponent = function (Component, elementToMount, options) {
    return exports.mountWithProvider(elementToMount, options).find(Component);
};
exports.mountWithProviderAndGetComponent = mountWithProviderAndGetComponent;
