// @fluentui/react-component-nesting-registry/dist/commonjs/hooks/useNestingRoot.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.useNestingRoot = void 0;

var React = _interopRequireWildcard(require("react"));

var _NestingContext = require("../NestingContext");

var _RefStack = require("../utils/RefStack");

var registrySet = new _RefStack.RefStack(); // These hooks are not used currently

/* eslint-disable */

var useNestingRoot = function useNestingRoot() {
  var _React$useState = React.useState(registrySet),
      registry = _React$useState[0];

  var parentRef = React.useRef(null);
  var nestedProps = React.useMemo(function () {
    return {
      value: {
        getContextRefs: registry.getContextRefs,
        register: registry.register,
        unregister: registry.unregister
      }
    };
  }, []);
  var getRefs = React.useCallback(function () {
    return registry.getContextRefs(parentRef);
  }, []);
  React.useEffect(function () {
    registry.register(parentRef);
    return function () {
      return registry.unregister(parentRef);
    };
  }, []);
  return {
    NestedComponent: _NestingContext.NestingContext.Provider,
    nestedProps: nestedProps,
    getRefs: getRefs,
    isRoot: true,
    ref: parentRef
  };
};

exports.useNestingRoot = useNestingRoot;
