// reakit/lib/Popover/Popover.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-8f9a8751.js');
var createComponent = require('reakit-system/createComponent');
var createHook = require('reakit-system/createHook');
require('reakit-utils/misc');
require('react');
var hooks = require('reakit-utils/hooks');
require('reakit-utils/dom');
var reakitWarning = require('reakit-warning');
require('reakit-utils/events');
require('reakit-utils/focus');
require('../Role/Role.js');
var useCreateElement = require('reakit-system/useCreateElement');
require('../__keys-f41a441b.js');
require('../Disclosure/DisclosureContent.js');
require('react-dom');
require('../Portal/Portal.js');
require('../MenuContext-8c9dd060.js');
var Dialog_Dialog = require('../Dialog/Dialog.js');
require('body-scroll-lock');
require('../DialogBackdropContext-b43e21d7.js');
require('../__keys-0c8e6398.js');
var __keys = require('../__keys-eddd3051.js');

var usePopover = createHook.createHook({
  name: "Popover",
  compose: Dialog_Dialog.useDialog,
  keys: __keys.POPOVER_KEYS,
  useOptions: function useOptions(_ref) {
    var _ref$modal = _ref.modal,
        modal = _ref$modal === void 0 ? false : _ref$modal,
        options = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_ref, ["modal"]);

    return _rollupPluginBabelHelpers._objectSpread2({
      modal: modal
    }, options);
  },
  useProps: function useProps(options, _ref2) {
    var htmlRef = _ref2.ref,
        htmlStyle = _ref2.style,
        htmlProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_ref2, ["ref", "style"]);

    return _rollupPluginBabelHelpers._objectSpread2({
      ref: hooks.useForkRef(options.unstable_popoverRef, htmlRef),
      style: _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, options.unstable_popoverStyles), htmlStyle)
    }, htmlProps);
  }
});
var Popover = createComponent.createComponent({
  as: "div",
  useHook: usePopover,
  useCreateElement: function useCreateElement$1(type, props, children) {
    process.env.NODE_ENV !== "production" ? reakitWarning.useWarning(!props["aria-label"] && !props["aria-labelledby"], "You should provide either `aria-label` or `aria-labelledby` props.", "See https://reakit.io/docs/popover") : void 0;
    return useCreateElement.useCreateElement(type, props, children);
  }
});

exports.Popover = Popover;
exports.usePopover = usePopover;
