// @fluentui/react-northstar/dist/commonjs/components/Debug/DebugLine.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.DebugLine = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var React = _interopRequireWildcard(require("react"));

var DebugLine = function DebugLine(_ref) {
  var active = _ref.active,
      _ref$indent = _ref.indent,
      indent = _ref$indent === void 0 ? 0 : _ref$indent,
      actionable = _ref.actionable,
      children = _ref.children,
      style = _ref.style,
      badge = _ref.badge,
      rest = (0, _objectWithoutPropertiesLoose2.default)(_ref, ["active", "indent", "actionable", "children", "style", "badge"]);
  return /*#__PURE__*/React.createElement("a", (0, _extends2.default)({
    style: Object.assign({
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      paddingLeft: indent * 2 + "ch",
      outline: 0
    }, actionable && {
      color: 'cornflowerblue',
      cursor: 'pointer'
    }, active && {
      background: 'rgba(255, 255, 255, 0.1)'
    }, style)
  }, rest), children, badge && /*#__PURE__*/React.createElement("span", {
    style: {
      padding: '0 4px',
      fontSize: 10,
      opacity: 0.75
    }
  }, badge));
};

exports.DebugLine = DebugLine;
