// @fluentui/react-northstar/dist/commonjs/components/Debug/ScrollToBottom.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.ScrollToBottom = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var React = _interopRequireWildcard(require("react"));

var ScrollToBottom = /*#__PURE__*/function (_React$Component) {
  (0, _inheritsLoose2.default)(ScrollToBottom, _React$Component);

  function ScrollToBottom() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    _this.ref = /*#__PURE__*/React.createRef();
    return _this;
  }

  var _proto = ScrollToBottom.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.scrollToBottom();
  };

  _proto.scrollToBottom = function scrollToBottom() {
    this.ref.current.scrollTo({
      behavior: 'smooth',
      top: 999999
    });
  };

  _proto.render = function render() {
    return /*#__PURE__*/React.createElement("div", (0, _extends2.default)({
      ref: this.ref
    }, this.props));
  };

  return ScrollToBottom;
}(React.Component);

exports.ScrollToBottom = ScrollToBottom;
