// @fluentui/react-northstar/dist/dts/src/components/Input/InputLabel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InputLabel = exports.inputLabelClassName = void 0;
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var utils_1 = require("../../utils");
var Box_1 = require("../Box/Box");
var PropTypes = require("prop-types");
exports.inputLabelClassName = 'ui-input__label';
/**
 * An InputLabel provides a slot for label in the Input.
 */
exports.InputLabel = react_bindings_1.compose(Box_1.Box, {
    className: exports.inputLabelClassName,
    displayName: 'InputLabel',
    overrideStyles: true,
    mapPropsToStylesProps: function (_a) {
        var labelPosition = _a.labelPosition, required = _a.required, hasValue = _a.hasValue;
        return ({
            labelPosition: labelPosition,
            required: required,
            hasValue: hasValue,
        });
    },
    handledProps: ['required', 'labelPosition', 'hasValue'],
    shorthandConfig: {
        mappedProp: 'content',
    },
});
exports.InputLabel.defaultProps = {
    as: 'label',
};
exports.InputLabel.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { labelPosition: PropTypes.oneOf(['inline', 'above', 'inside']), required: PropTypes.bool, hasValue: PropTypes.bool });
