// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Datepicker/datepickerCalendarCellButtonStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.datepickerCalendarCellButtonStyles = void 0;
var tslib_1 = require("tslib");
var getBorderFocusStyles_1 = require("../../getBorderFocusStyles");
exports.datepickerCalendarCellButtonStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables, theme = _a.theme;
        var siteVariables = theme.siteVariables;
        var borderWidth = siteVariables.borderWidth;
        var borderFocusStyles = getBorderFocusStyles_1.getBorderFocusStyles({
            variables: siteVariables,
            borderPadding: borderWidth,
        });
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ textAlign: 'center', height: '100%', width: '100%', cursor: 'pointer', border: v.calendarCellBorder, padding: v.calendarCellPadding, margin: v.calendarCellMargin, display: 'table-cell', position: 'relative' }, borderFocusStyles), { backgroundColor: v.calendarCellBackgroundColor, color: v.calendarCellColor, ':hover': {
                backgroundColor: v.calendarCellHoverBackgroundColor,
                color: v.calendarCellHoverColor,
            } }), (p.quiet && {
            color: v.calendarCellQuietColor,
        })), (p.selected && {
            color: v.calendarCellSelectedColor,
            backgroundColor: v.calendarCellSelectedBackgroundColor,
        })), (p.disabled && {
            color: v.calendarCellDisabledColor,
            cursor: 'default',
            backgroundColor: v.calendarCellDisabledBackgroundColor,
            ':hover': {},
        })), (p.today && {
            backgroundColor: v.calendarCellTodayBackgroundColor,
            color: v.calendarCellTodayColor,
            borderRadius: v.calendarCellTodayBorderRadius,
        }));
    },
};
