// react-cosmos-shared2/dist/url.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stringifyPlaygroundUrlQuery = stringifyPlaygroundUrlQuery;
exports.parsePlaygroundUrlQuery = parsePlaygroundUrlQuery;
exports.stringifyRendererUrlQuery = stringifyRendererUrlQuery;
exports.parseRendererUrlQuery = parseRendererUrlQuery;
exports.parseUrlQuery = parseUrlQuery;
exports.stringifyUrlQuery = stringifyUrlQuery;

var _queryString = _interopRequireDefault(require("query-string"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function stringifyPlaygroundUrlQuery(urlParams) {
  var encodedUrlParams = {};

  if (urlParams.fixtureId) {
    encodedUrlParams.fixtureId = JSON.stringify(urlParams.fixtureId);
  }

  return _queryString.default.stringify(encodedUrlParams);
}

function parsePlaygroundUrlQuery(query) {
  var encodedUrlParams = parseUrlQuery(query);
  var decoded = {};

  if (encodedUrlParams.fixtureId) {
    decoded.fixtureId = JSON.parse(encodedUrlParams.fixtureId);
  }

  return decoded;
}

function stringifyRendererUrlQuery(urlParams) {
  var encodedUrlParams = {};

  if (urlParams._fixtureId) {
    encodedUrlParams._fixtureId = JSON.stringify(urlParams._fixtureId);
  }

  return _queryString.default.stringify(encodedUrlParams);
}

function parseRendererUrlQuery(query) {
  var encodedUrlParams = parseUrlQuery(query);
  var decoded = {};

  if (encodedUrlParams._fixtureId) {
    decoded._fixtureId = JSON.parse(encodedUrlParams._fixtureId);
  }

  return decoded;
}

function parseUrlQuery(query) {
  return _queryString.default.parse(query);
}

function stringifyUrlQuery(params) {
  return _queryString.default.stringify(params);
}
