// @fluentui/react-northstar/dist/commonjs/components/Breadcrumb/BreadcrumbDivider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.BreadcrumbDivider = exports.breadcrumbDividerClassName = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactBindings = require("@fluentui/react-bindings");

var _utils = require("../../utils");

var _accessibility = require("@fluentui/accessibility");

var breadcrumbDividerClassName = 'ui-breadcrumb__divider';
/**
 * BreadcrumbDivider divides BreadcrumbItem components within Breadcrumb
 */

exports.breadcrumbDividerClassName = breadcrumbDividerClassName;
var BreadcrumbDivider = (0, _reactBindings.compose)(function (props, ref, composeOptions) {
  var context = (0, _reactBindings.useFluentContext)();

  var _useTelemetry = (0, _reactBindings.useTelemetry)(composeOptions.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var accessibility = props.accessibility,
      children = props.children,
      content = props.content,
      className = props.className,
      design = props.design,
      styles = props.styles,
      variables = props.variables;
  var getA11yProps = (0, _reactBindings.useAccessibility)(accessibility, {
    debugName: composeOptions.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(composeOptions.displayName, {
    className: composeOptions.className,
    composeOptions: composeOptions,
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl,
    unstable_props: props
  }),
      classes = _useStyles.classes;

  var unhandledProps = (0, _reactBindings.useUnhandledProps)(composeOptions.handledProps, props);
  var ElementType = (0, _reactBindings.getElementType)(props);
  var result = /*#__PURE__*/React.createElement(ElementType, getA11yProps('root', Object.assign({
    className: classes.root,
    ref: ref
  }, unhandledProps)), (0, _reactBindings.childrenExist)(children) ? children : content);
  setEnd();
  return result;
}, {
  className: breadcrumbDividerClassName,
  displayName: 'BreadcrumbDivider',
  handledProps: ['accessibility', 'as', 'children', 'className', 'content', 'design', 'styles', 'variables']
});
exports.BreadcrumbDivider = BreadcrumbDivider;
BreadcrumbDivider.defaultProps = {
  as: 'span',
  children: '/',
  accessibility: _accessibility.breadcrumbDividerBehavior
};
BreadcrumbDivider.propTypes = _utils.commonPropTypes.createCommon();
