// @fluentui/accessibility/dist/dts/src/behaviors/Pill/pillBehavior.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pillBehavior = void 0;
var tslib_1 = require("tslib");
var keyboard_key_1 = require("../../keyboard-key");
var attributes_1 = require("../../attributes");
var pillBehavior = function (p) {
    var _a;
    return ({
        attributes: {
            root: tslib_1.__assign((_a = { role: p.actionable ? 'button' : 'none', tabIndex: p.actionable ? 0 : -1 }, _a[attributes_1.IS_FOCUSABLE_ATTRIBUTE] = p.actionable || p.role === 'option', _a), (p.selectable && {
                'aria-selected': p.selected,
            })),
        },
        keyActions: {
            root: tslib_1.__assign({}, (p.actionable && {
                performDismiss: {
                    keyCombinations: [{ keyCode: keyboard_key_1.keyboardKey.Delete }, { keyCode: keyboard_key_1.keyboardKey.Backspace }],
                },
                performClick: {
                    keyCombinations: [{ keyCode: keyboard_key_1.keyboardKey.Enter }, { keyCode: keyboard_key_1.SpacebarKey }],
                },
            })),
        },
    });
};
exports.pillBehavior = pillBehavior;
