// @fluentui/react-northstar/dist/commonjs/components/Avatar/AvatarLabel.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.AvatarLabel = exports.avatarlabelClassName = void 0;

var _reactBindings = require("@fluentui/react-bindings");

var customPropTypes = _interopRequireWildcard(require("@fluentui/react-proptypes"));

var PropTypes = _interopRequireWildcard(require("prop-types"));

var React = _interopRequireWildcard(require("react"));

var _utils = require("../../utils");

var _Label = require("../Label/Label");

var avatarlabelClassName = _Label.labelClassName;
/**
 * A AvatarLabel allows user to classify content.
 */

exports.avatarlabelClassName = avatarlabelClassName;

var AvatarLabel = function AvatarLabel(props) {
  var context = (0, _reactBindings.useFluentContext)();

  var _useTelemetry = (0, _reactBindings.useTelemetry)(AvatarLabel.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var accessibility = props.accessibility,
      children = props.children,
      className = props.className,
      content = props.content,
      design = props.design,
      styles = props.styles,
      variables = props.variables,
      square = props.square,
      size = props.size,
      circular = props.circular;
  var getA11Props = (0, _reactBindings.useAccessibility)(accessibility, {
    debugName: AvatarLabel.displayName,
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(AvatarLabel.displayName, {
    className: avatarlabelClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        square: square,
        size: size,
        circular: circular
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(AvatarLabel.handledProps, props);
  var element = /*#__PURE__*/React.createElement(ElementType, getA11Props('root', Object.assign({
    className: classes.root
  }, _utils.rtlTextContainer.getAttributes({
    forElements: [children]
  }), unhandledProps)), (0, _utils.childrenExist)(children) ? children : content);
  setEnd();
  return element;
};

exports.AvatarLabel = AvatarLabel;
AvatarLabel.displayName = 'AvatarLabel';
AvatarLabel.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon(), {
  square: PropTypes.bool,
  size: customPropTypes.size,
  circular: PropTypes.bool
});
AvatarLabel.handledProps = Object.keys(AvatarLabel.propTypes);
AvatarLabel.defaultProps = {
  as: 'span'
};
AvatarLabel.shorthandConfig = {
  mappedProp: 'content'
};
