// @fluentui/react-northstar/dist/commonjs/components/Avatar/AvatarImage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

exports.__esModule = true;
exports.AvatarImage = exports.avatarImageClassName = void 0;

var _accessibility = require("@fluentui/accessibility");

var _reactBindings = require("@fluentui/react-bindings");

var PropTypes = _interopRequireWildcard(require("prop-types"));

var React = _interopRequireWildcard(require("react"));

var _utils = require("../../utils");

var _Image = require("../Image/Image");

var avatarImageClassName = _Image.imageClassName;
/**
 * An AvatarImage is a graphic representation used by Avatar.
 */

exports.avatarImageClassName = avatarImageClassName;

var AvatarImage = function AvatarImage(props) {
  var context = (0, _reactBindings.useFluentContext)();

  var _useTelemetry = (0, _reactBindings.useTelemetry)(AvatarImage.displayName, context.telemetry),
      setStart = _useTelemetry.setStart,
      setEnd = _useTelemetry.setEnd;

  setStart();
  var accessibility = props.accessibility,
      alt = props.alt,
      ariaLabel = props['aria-label'],
      avatar = props.avatar,
      circular = props.circular,
      className = props.className,
      design = props.design,
      fluid = props.fluid,
      styles = props.styles,
      variables = props.variables,
      size = props.size;
  var getA11Props = (0, _reactBindings.useAccessibility)(accessibility, {
    debugName: AvatarImage.displayName,
    mapPropsToBehavior: function mapPropsToBehavior() {
      return {
        alt: alt,
        'aria-label': ariaLabel
      };
    },
    rtl: context.rtl
  });

  var _useStyles = (0, _reactBindings.useStyles)(AvatarImage.displayName, {
    className: avatarImageClassName,
    mapPropsToStyles: function mapPropsToStyles() {
      return {
        avatar: avatar,
        circular: circular,
        fluid: fluid,
        size: size
      };
    },
    mapPropsToInlineStyles: function mapPropsToInlineStyles() {
      return {
        className: className,
        design: design,
        styles: styles,
        variables: variables
      };
    },
    rtl: context.rtl
  }),
      classes = _useStyles.classes;

  var ElementType = (0, _reactBindings.getElementType)(props);
  var unhandledProps = (0, _reactBindings.useUnhandledProps)(AvatarImage.handledProps, props);
  var result = /*#__PURE__*/React.createElement(ElementType, getA11Props('root', Object.assign({
    className: classes.root
  }, unhandledProps)));
  setEnd();
  return result;
};

exports.AvatarImage = AvatarImage;
AvatarImage.displayName = 'AvatarImage';
AvatarImage.defaultProps = {
  as: 'img',
  accessibility: _accessibility.imageBehavior
};
AvatarImage.propTypes = Object.assign({}, _utils.commonPropTypes.createCommon({
  children: false,
  content: false
}), {
  avatar: PropTypes.bool,
  circular: PropTypes.bool,
  fluid: PropTypes.bool
});
AvatarImage.shorthandConfig = {
  mappedProp: 'src',
  allowsJSX: false
};
AvatarImage.handledProps = Object.keys(AvatarImage.propTypes);
AvatarImage.create = (0, _utils.createShorthandFactory)({
  Component: AvatarImage,
  mappedProp: 'src',
  allowsJSX: false
});
