// @fluentui/react-northstar/dist/dts/src/components/Button/ButtonContent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ButtonContent = exports.buttonContentClassName = void 0;
var tslib_1 = require("tslib");
var react_bindings_1 = require("@fluentui/react-bindings");
var customPropTypes = require("@fluentui/react-proptypes");
var utils_1 = require("../../utils");
var Box_1 = require("../Box/Box");
exports.buttonContentClassName = 'ui-button__content';
/**
 * A ButtonContent allows a user to have a dedicated component that can be targeted from the theme.
 */
exports.ButtonContent = react_bindings_1.compose(Box_1.Box, {
    className: exports.buttonContentClassName,
    displayName: 'ButtonContent',
    mapPropsToStylesProps: function (props) { return ({ size: props.size }); },
    handledProps: ['size'],
    overrideStyles: true,
    shorthandConfig: {
        mappedProp: 'content',
    },
});
exports.ButtonContent.defaultProps = {
    as: 'span',
};
exports.ButtonContent.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { size: customPropTypes.size });
