// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Menu/menuStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.menuStyles = void 0;
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
var colors_1 = require("../../colors");
exports.menuStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables, theme = _a.theme;
        var iconOnly = p.iconOnly, fluid = p.fluid, pointing = p.pointing, pills = p.pills, primary = p.primary, underlined = p.underlined, vertical = p.vertical, submenu = p.submenu;
        var colors = colors_1.getColorScheme(v.colorScheme, null, primary);
        var siteVariables = theme.siteVariables;
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ display: 'flex', minHeight: utils_1.pxToRem(24), margin: 0, padding: 0, color: v.color, backgroundColor: v.backgroundColor || 'inherit', listStyleType: 'none' }, (iconOnly && { alignItems: 'center' })), (vertical && tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ flexDirection: 'column', backgroundColor: v.verticalBackgroundColor, padding: utils_1.pxToRem(8) + " 0" }, (submenu && {
            boxShadow: v.verticalBoxShadow,
        })), (!fluid && !submenu && { width: 'fit-content' })), (iconOnly && {
            display: 'inline-block',
            width: 'auto',
        })))), (!pills &&
            !iconOnly &&
            !(pointing && vertical) &&
            !underlined && {
            // primary has hardcoded grey border color
            border: v.borderWidth + " solid " + (primary ? v.primaryBorderColor : v.borderColor || colors.border),
            borderRadius: siteVariables.borderRadiusMedium,
        })), (underlined && {
            borderBottom: v.underlinedBottomBorderWidth + " solid " + v.underlinedBorderColor,
        }));
    },
};
