// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Menu/menuItemIndicatorStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.menuItemIndicatorStyles = void 0;
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
var submenuIndicatorDirection_1 = require("./submenuIndicatorDirection");
exports.menuItemIndicatorStyles = {
    root: function (_a) {
        var p = _a.props, v = _a.variables;
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({ position: 'relative', float: 'right', left: utils_1.pxToRem(12), userSelect: 'none', marginRight: utils_1.pxToRem(4) }, (p.inSubmenu && {
            position: 'absolute',
            top: utils_1.pxToRem(6),
            right: utils_1.pxToRem(2),
            left: 'unset',
        })), { content: '" "', display: 'block', overflow: 'hidden', height: utils_1.pxToRem(16), width: utils_1.pxToRem(16), backgroundSize: utils_1.pxToRem(16), color: v.indicatorColor }), (p.active && tslib_1.__assign({ color: v.activeIndicatorColor }, (p.primary && tslib_1.__assign({ color: v.activePrimaryIndicatorColor }, (p.vertical && {
            color: v.activePrimaryVerticalIndicatorColor,
        })))))), (p.underlined && { color: v.indicatorColor })), (p.iconOnly && { color: v.indicatorColor })), submenuIndicatorDirection_1.submenuIndicatorDirection(p.vertical));
    },
};
