// @fluentui/react-northstar/dist/dts/src/themes/teams/components/Form/formFieldStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.formFieldStyles = void 0;
var tslib_1 = require("tslib");
var utils_1 = require("../../../../utils");
exports.formFieldStyles = {
    root: function (_a) {
        var props = _a.props, variables = _a.variables;
        return ({});
    },
    label: function (_a) {
        var props = _a.props;
        var type = props.type, inline = props.inline, required = props.required;
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, ((!type || (type !== 'radio' && type !== 'checkbox')) && {
            display: 'block',
            marginBottom: utils_1.pxToRem(4),
        })), (inline && { marginRight: utils_1.pxToRem(10), display: 'inline' })), (required && {
            '::after': {
                content: '"*"',
            },
        }));
    },
    control: function (_a) {
        var props = _a.props, v = _a.variables;
        var type = props.type;
        return tslib_1.__assign({}, (type &&
            (type === 'radio' || type === 'checkbox') && {
            marginRight: utils_1.pxToRem(10),
        }));
    },
    message: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign(tslib_1.__assign({}, (p.hasErrorMessage && { color: v.colorScheme.red.foreground })), { display: 'block', paddingLeft: v.messagePaddingLeft }));
    },
    icon: function (_a) {
        var p = _a.props, v = _a.variables;
        return (tslib_1.__assign({}, (p.hasErrorMessage && { color: v.colorScheme.red.foreground })));
    },
};
