// @fluentui/react-northstar/dist/dts/src/components/Toolbar/ToolbarCustomItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolbarCustomItem = exports.toolbarCustomItemClassName = void 0;
var tslib_1 = require("tslib");
var accessibility_1 = require("@fluentui/accessibility");
var react_bindings_1 = require("@fluentui/react-bindings");
var _ = require("lodash");
var PropTypes = require("prop-types");
var React = require("react");
var utils_1 = require("../../utils");
var toolbarVariablesContext_1 = require("./toolbarVariablesContext");
exports.toolbarCustomItemClassName = 'ui-toolbar__customitem';
/**
 * A ToolbarCustomItem renders Toolbar item as a non-actionable `div` with custom content inside.
 */
exports.ToolbarCustomItem = react_bindings_1.compose(function (props, ref, composeOptions) {
    var _a;
    var context = react_bindings_1.useFluentContext();
    var _b = react_bindings_1.useTelemetry(composeOptions.displayName, context.telemetry), setStart = _b.setStart, setEnd = _b.setEnd;
    setStart();
    var accessibility = props.accessibility, children = props.children, className = props.className, content = props.content, design = props.design, fitted = props.fitted, focusable = props.focusable, styles = props.styles, variables = props.variables;
    var parentVariables = React.useContext(toolbarVariablesContext_1.ToolbarVariablesContext);
    var getA11yProps = react_bindings_1.useAccessibility(accessibility, {
        debugName: composeOptions.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(composeOptions.displayName, {
        className: composeOptions.className,
        composeOptions: composeOptions,
        mapPropsToStyles: function () { return ({ fitted: fitted }); },
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: react_bindings_1.mergeVariablesOverrides(parentVariables, variables),
        }); },
        rtl: context.rtl,
        unstable_props: props,
    }).classes;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(composeOptions.handledProps, props);
    var handleBlur = function (e) {
        if (focusable) {
            _.invoke(props, 'onBlur', e, props);
        }
    };
    var handleFocus = function (e) {
        if (focusable) {
            _.invoke(props, 'onFocus', e, props);
        }
    };
    var element = (React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign(tslib_1.__assign((_a = {}, _a[accessibility_1.IS_FOCUSABLE_ATTRIBUTE] = focusable, _a), unhandledProps), { className: classes.root, onBlur: handleBlur, onFocus: handleFocus, ref: ref }))), utils_1.childrenExist(children) ? children : content));
    setEnd();
    return element;
}, {
    className: exports.toolbarCustomItemClassName,
    displayName: 'ToolbarCustomItem',
    shorthandConfig: { mappedProp: 'content' },
    handledProps: [
        'accessibility',
        'as',
        'children',
        'className',
        'content',
        'design',
        'fitted',
        'focusable',
        'onBlur',
        'onFocus',
        'styles',
        'variables',
    ],
});
exports.ToolbarCustomItem.propTypes = tslib_1.__assign(tslib_1.__assign({}, utils_1.commonPropTypes.createCommon()), { fitted: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.oneOf(['horizontally', 'vertically']),
    ]), focusable: PropTypes.bool, onFocus: PropTypes.func, onBlur: PropTypes.func });
