// @fluentui/react-northstar/dist/dts/src/components/Toolbar/ToolbarMenuDivider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolbarMenuDivider = exports.toolbarMenuDividerClassName = void 0;
var tslib_1 = require("tslib");
var accessibility_1 = require("@fluentui/accessibility");
var react_bindings_1 = require("@fluentui/react-bindings");
var React = require("react");
var utils_1 = require("../../utils");
var toolbarVariablesContext_1 = require("./toolbarVariablesContext");
exports.toolbarMenuDividerClassName = 'ui-toolbar__menudivider';
/**
 * A ToolbarMenuDivider adds non-actionable separator between items of ToolbarMenu.
 */
exports.ToolbarMenuDivider = react_bindings_1.compose(function (props, ref, composeOptions) {
    var context = react_bindings_1.useFluentContext();
    var _a = react_bindings_1.useTelemetry(composeOptions.displayName, context.telemetry), setStart = _a.setStart, setEnd = _a.setEnd;
    setStart();
    var accessibility = props.accessibility, className = props.className, design = props.design, styles = props.styles, variables = props.variables;
    var parentVariables = React.useContext(toolbarVariablesContext_1.ToolbarVariablesContext);
    var getA11yProps = react_bindings_1.useAccessibility(accessibility, {
        debugName: composeOptions.displayName,
        rtl: context.rtl,
    });
    var classes = react_bindings_1.useStyles(composeOptions.displayName, {
        className: composeOptions.className,
        composeOptions: composeOptions,
        mapPropsToInlineStyles: function () { return ({
            className: className,
            design: design,
            styles: styles,
            variables: react_bindings_1.mergeVariablesOverrides(parentVariables, variables),
        }); },
        rtl: context.rtl,
        unstable_props: props,
    }).classes;
    var ElementType = react_bindings_1.getElementType(props);
    var unhandledProps = react_bindings_1.useUnhandledProps(composeOptions.handledProps, props);
    var element = React.createElement(ElementType, tslib_1.__assign({}, getA11yProps('root', tslib_1.__assign(tslib_1.__assign({}, unhandledProps), { className: classes.root, ref: ref }))));
    setEnd();
    return element;
}, {
    className: exports.toolbarMenuDividerClassName,
    displayName: 'ToolbarMenuDivider',
    shorthandConfig: { mappedProp: 'content' },
    handledProps: ['accessibility', 'as', 'children', 'className', 'content', 'design', 'styles', 'variables'],
});
exports.ToolbarMenuDivider.propTypes = utils_1.commonPropTypes.createCommon();
exports.ToolbarMenuDivider.defaultProps = {
    as: 'li',
    accessibility: accessibility_1.toolbarMenuDividerBehavior,
};
