// reakit/lib/Tab/TabState.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-8f9a8751.js');
require('reakit-utils/misc');
var React = require('react');
var hooks = require('reakit-utils/hooks');
require('reakit-utils/dom');
require('../reverse-4756a49e.js');
require('../getCurrentId-eade2850.js');
require('../findEnabledItemById-03112678.js');
require('../Id/IdProvider.js');
require('../Id/IdState.js');
var Composite_CompositeState = require('../Composite/CompositeState.js');

function useTabState(initialState) {
  if (initialState === void 0) {
    initialState = {};
  }

  var _useInitialValue = hooks.useInitialValue(initialState),
      initialSelectedId = _useInitialValue.selectedId,
      _useInitialValue$loop = _useInitialValue.loop,
      loop = _useInitialValue$loop === void 0 ? true : _useInitialValue$loop,
      _useInitialValue$manu = _useInitialValue.manual,
      manual = _useInitialValue$manu === void 0 ? false : _useInitialValue$manu,
      sealed = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_useInitialValue, ["selectedId", "loop", "manual"]);

  var composite = Composite_CompositeState.useCompositeState(_rollupPluginBabelHelpers._objectSpread2({
    loop: loop,
    currentId: initialSelectedId
  }, sealed));
  var panels = Composite_CompositeState.useCompositeState();

  var _React$useState = React.useState(initialSelectedId),
      selectedId = _React$useState[0],
      setSelectedId = _React$useState[1];

  var select = React.useCallback(function (id) {
    composite.move(id);
    setSelectedId(id);
  }, [composite.move]); // If selectedId is not set, use the currentId. It's still possible to have
  // no selected tab with useTabState({ selectedId: null });

  React.useEffect(function () {
    if (selectedId === null) return;
    var selectedItem = composite.items.find(function (item) {
      return item.id === selectedId;
    });
    if (selectedItem) return;

    if (composite.currentId) {
      setSelectedId(composite.currentId);
    }
  }, [selectedId, composite.items, composite.currentId]);
  return _rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({}, composite), {}, {
    selectedId: selectedId,
    panels: panels.items,
    manual: manual,
    select: select,
    setSelectedId: setSelectedId,
    registerPanel: React.useCallback(function (panel) {
      return panels.registerItem(panel);
    }, [panels.registerItem]),
    unregisterPanel: React.useCallback(function (id) {
      return panels.unregisterItem(id);
    }, [panels.unregisterItem])
  });
}

exports.useTabState = useTabState;
