// @fluentui/react-bindings/dist/dts/src/hooks/useStyles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useStyles = void 0;
var tslib_1 = require("tslib");
var React = require("react");
var context_1 = require("../context");
var getStyles_1 = require("../styles/getStyles");
var useStyles = function (displayName, options) {
    var _a;
    var context = context_1.useFluentContext();
    var _b = options.className, className = _b === void 0 ? process.env.NODE_ENV === 'production' ? '' : 'no-classname-🙉' : _b, composeOptions = options.composeOptions, _c = options.mapPropsToStyles, mapPropsToStyles = _c === void 0 ? function () { return ({}); } : _c, _d = options.mapPropsToInlineStyles, mapPropsToInlineStyles = _d === void 0 ? function () { return ({}); } : _d, _e = options.unstable_props, unstable_props = _e === void 0 ? {} : _e, _f = options.rtl, rtl = _f === void 0 ? false : _f;
    var componentStylesProps = mapPropsToStyles();
    // `composeProps` should include all props including stylesProps as they can contain state
    var composeProps = tslib_1.__assign(tslib_1.__assign({}, unstable_props), componentStylesProps);
    var composeStylesProps = (_a = composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.mapPropsToStylesPropsChain) === null || _a === void 0 ? void 0 : _a.reduce(function (acc, fn) { return (tslib_1.__assign(tslib_1.__assign({}, acc), fn(composeProps))); }, {});
    // Stores debug information for component.
    var debug = React.useRef({ fluentUIDebug: null });
    var _g = getStyles_1.getStyles({
        // Input values
        allDisplayNames: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.displayNames) || [displayName],
        className: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.className) || className,
        primaryDisplayName: (composeOptions === null || composeOptions === void 0 ? void 0 : composeOptions.displayName) || displayName,
        componentProps: tslib_1.__assign(tslib_1.__assign({}, componentStylesProps), composeStylesProps),
        inlineStylesProps: mapPropsToInlineStyles(),
        // Context values
        disableAnimations: context.disableAnimations,
        renderer: context.renderer,
        rtl: rtl,
        saveDebug: function (fluentUIDebug) { return (debug.current = { fluentUIDebug: fluentUIDebug }); },
        theme: context.theme,
        performance: context.performance,
        telemetry: context.telemetry,
    }), classes = _g.classes, resolvedStyles = _g.styles;
    return { classes: classes, styles: resolvedStyles };
};
exports.useStyles = useStyles;
