// react-cosmos-shared2/dist/FixtureLoader/FixtureCapture/classState/useReadClassState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useReadClassState = useReadClassState;

var _lodash = require("lodash");

var _react = _interopRequireDefault(require("react"));

var _fixtureState = require("../../../fixtureState");

var _FixtureContext = require("../../FixtureContext");

var _findRelevantElementPaths = require("../shared/findRelevantElementPaths");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

// How often to check the state of the loaded component and update the fixture
// state if it changed
var REFRESH_INTERVAL = 200;

function useReadClassState(fixture, decoratorId, elRefs) {
  var elPaths = (0, _findRelevantElementPaths.findRelevantElementPaths)(fixture);

  var _React$useContext = _react.default.useContext(_FixtureContext.FixtureContext),
      fixtureState = _React$useContext.fixtureState,
      setFixtureState = _React$useContext.setFixtureState;

  var timeoutId = _react.default.useRef(null);

  _react.default.useEffect(function () {
    // The check should run even if no element paths are found at mount, because
    // the fixture can change during the lifecycle of a FixtureCapture instance
    // and the updated fixture might contain elements of stateful components
    scheduleStateCheck();
    return function () {
      if (timeoutId.current) {
        clearTimeout(timeoutId.current);
      }
    };
  });

  function scheduleStateCheck() {
    // Is there a better way to listen to component state changes?
    timeoutId.current = window.setTimeout(checkState, REFRESH_INTERVAL);
  }

  function checkState() {
    var fixtureStateChangeScheduled = false;
    Object.keys(elRefs.current).map( /*#__PURE__*/function () {
      var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(elPath) {
        var state, elementId, fsClassState;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                if (!(elPaths.indexOf(elPath) === -1)) {
                  _context.next = 2;
                  break;
                }

                throw new Error("[FixtureCapture] Child ref exists for missing element path \"".concat(elPath, "\""));

              case 2:
                state = elRefs.current[elPath].state;
                elementId = {
                  decoratorId: decoratorId,
                  elPath: elPath
                };
                fsClassState = (0, _fixtureState.findFixtureStateClassState)(fixtureState, elementId);

                if (fsClassState && state && !doesFixtureStateMatchClassState(fsClassState, state)) {
                  fixtureStateChangeScheduled = true;
                  setFixtureState(function (prevFs) {
                    return _objectSpread(_objectSpread({}, prevFs), {}, {
                      classState: (0, _fixtureState.updateFixtureStateClassState)({
                        fixtureState: prevFs,
                        elementId: elementId,
                        values: (0, _fixtureState.createValues)(state)
                      })
                    });
                  });
                }

              case 6:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      return function (_x) {
        return _ref.apply(this, arguments);
      };
    }());

    if (!fixtureStateChangeScheduled) {
      scheduleStateCheck();
    }
  }
}

function doesFixtureStateMatchClassState(fsClassState, state) {
  return (0, _lodash.isEqual)(state, (0, _fixtureState.extendWithValues)(state, fsClassState.values));
}
