// reakit/lib/Combobox/ComboboxListState.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-8f9a8751.js');
require('reakit-utils/misc');
require('react');
var hooks = require('reakit-utils/hooks');
require('reakit-utils/dom');
require('../reverse-4756a49e.js');
require('../getCurrentId-eade2850.js');
require('../findEnabledItemById-03112678.js');
require('../Id/IdProvider.js');
require('../Id/IdState.js');
var Composite_CompositeState = require('../Composite/CompositeState.js');
var ComboboxBaseState = require('../ComboboxBaseState-cfd43319.js');

function unstable_useComboboxListState(initialState) {
  if (initialState === void 0) {
    initialState = {};
  }

  var _useInitialValue = hooks.useInitialValue(initialState),
      _useInitialValue$curr = _useInitialValue.currentId,
      currentId = _useInitialValue$curr === void 0 ? null : _useInitialValue$curr,
      _useInitialValue$orie = _useInitialValue.orientation,
      orientation = _useInitialValue$orie === void 0 ? "vertical" : _useInitialValue$orie,
      _useInitialValue$loop = _useInitialValue.loop,
      loop = _useInitialValue$loop === void 0 ? true : _useInitialValue$loop,
      sealed = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_useInitialValue, ["currentId", "orientation", "loop"]);

  var composite = Composite_CompositeState.useCompositeState(_rollupPluginBabelHelpers._objectSpread2(_rollupPluginBabelHelpers._objectSpread2({
    currentId: currentId,
    orientation: orientation,
    loop: loop
  }, sealed), {}, {
    unstable_virtual: true,
    unstable_includesBaseElement: true
  }));
  return ComboboxBaseState.useComboboxBaseState(composite, sealed);
}

exports.unstable_useComboboxListState = unstable_useComboboxListState;
